diff -Nur kolourpaint/CMakeLists.txt kolourpaintn/CMakeLists.txt
--- kolourpaint/CMakeLists.txt	2015-07-03 22:45:29.000000000 +0000
+++ kolourpaintn/CMakeLists.txt	2015-07-04 00:07:46.567713693 +0000
@@ -1,5 +1,8 @@
 project(kolourpaint)
 
+# it's come from kde4 cmake files, but missing for kf5
+set(GENERIC_LIB_VERSION "5.11.0")
+set(GENERIC_LIB_SOVERSION "5")
 cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
 set(QT_MIN_VERSION "5.2.0")
 
