diff -Nur qca-2.1.0/unittest/CMakeLists.txt qca-2.1.0n/unittest/CMakeLists.txt
--- qca-2.1.0/unittest/CMakeLists.txt	2014-11-06 08:15:45.000000000 +0000
+++ qca-2.1.0n/unittest/CMakeLists.txt	2015-10-25 12:04:57.167498382 +0000
@@ -23,7 +23,7 @@
 add_subdirectory(keystore)
 add_subdirectory(macunittest)
 add_subdirectory(metatype)
-add_subdirectory(pgpunittest)
+#add_subdirectory(pgpunittest)
 add_subdirectory(pipeunittest)
 add_subdirectory(pkits)
 add_subdirectory(rsaunittest)
