diff -Nur qca-2.1.0/include/QtCrypto/qca_basic.h qca-2.1.0n/include/QtCrypto/qca_basic.h
--- qca-2.1.0/include/QtCrypto/qca_basic.h	2014-11-06 08:15:45.000000000 +0000
+++ qca-2.1.0n/include/QtCrypto/qca_basic.h	2015-08-13 15:35:08.098815116 +0000
@@ -34,6 +34,7 @@
 #define QCA_BASIC_H
 
 #include "qca_core.h"
+#include <QIODevice>
 
 // Qt5 comes with QStringLiteral for wrapping string literals, which Qt4 does
 // not have. It is needed if the headers are built with QT_NO_CAST_FROM_ASCII.
