diff --git a/src/declarativeimports/core/datasource.cpp b/src/declarativeimports/core/datasource.cpp
index 4fe5dc5..17356e3 100644
--- a/src/declarativeimports/core/datasource.cpp
+++ b/src/declarativeimports/core/datasource.cpp
@@ -267,4 +267,9 @@ void DataSource::updateSources()
     }
 }
 
+bool DataSource::isFileExists(QString path)
+{
+    return QFile::exists(path);
+}
+
 }
diff --git a/src/declarativeimports/core/datasource.h b/src/declarativeimports/core/datasource.h
index 507daf8..01cf3da 100644
--- a/src/declarativeimports/core/datasource.h
+++ b/src/declarativeimports/core/datasource.h
@@ -157,6 +157,8 @@ public:
      */
     Q_INVOKABLE void disconnectSource(const QString &source);
 
+    Q_INVOKABLE bool isFileExists(QString path);
+
 public Q_SLOTS:
     void dataUpdated(const QString &sourceName, const Plasma::DataEngine::Data &data);
     void modelChanged(const QString &sourceName, QAbstractItemModel *model);
