Index: plasma-framework-5.16.0/src/desktoptheme/breeze/metadata.desktop
===================================================================
--- plasma-framework-5.16.0.orig/src/desktoptheme/breeze/metadata.desktop
+++ plasma-framework-5.16.0/src/desktoptheme/breeze/metadata.desktop
@@ -50,7 +50,7 @@ X-KDE-PluginInfo-EnabledByDefault=true
 X-Plasma-API=5.0
 
 [Wallpaper]
-defaultWallpaperTheme=Next
+defaultWallpaperTheme=Carat
 defaultFileSuffix=.png
 defaultWidth=1920
 defaultHeight=1080
