diff --git a/src/declarativeimports/plasmacomponents/qml/private/TabBarLayout.qml b/src/declarativeimports/plasmacomponents/qml/private/TabBarLayout.qml
index 6bc6fc9..49f08a7 100644
--- a/src/declarativeimports/plasmacomponents/qml/private/TabBarLayout.qml
+++ b/src/declarativeimports/plasmacomponents/qml/private/TabBarLayout.qml
@@ -68,12 +68,11 @@ Item {
     property int minimumWidth: 0
     property int minimumHeight: 0
 
-    Component.onCompleted: priv.layoutChildren()
+    Component.onCompleted: layoutTimer.restart()
     onChildrenChanged: layoutTimer.restart()
     onWidthChanged: layoutTimer.restart()
     onHeightChanged: layoutTimer.restart()
 
-
     Keys.onPressed: {
         if (event.key == Qt.Key_Right || event.key == Qt.Key_Left) {
             if (event.key == Qt.Key_Right || priv.mirrored) {
@@ -90,9 +89,10 @@ Item {
 
     Timer {
         id: layoutTimer
-        interval: 150
+        interval: 10
         onTriggered: priv.layoutChildren()
     }
+
     MouseEventListener {
         anchors.fill: parent
         onWheelMoved: {
