diff -Nur plasma-framework-5.16.0/src/declarativeimports/calendar/qml/DayDelegate.qml plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/DayDelegate.qml
--- plasma-framework-5.16.0/src/declarativeimports/calendar/qml/DayDelegate.qml	2016-01-08 09:50:16.405671644 +0800
+++ plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/DayDelegate.qml	2016-01-08 09:51:50.397671633 +0800
@@ -125,4 +125,50 @@
             root.date = model
         }
     }
+
+    Lunar {
+        id: lunar
+        year: thisDate.getFullYear()
+        month: thisDate.getMonth()
+        day: model.label || dayNumber
+    }
+
+    Rectangle {
+         width: 5
+         height: width
+         x: 5
+         y: 5
+         color: theme.highlightColor
+         border.color: theme.highlightColor
+         border.width: 1
+         radius: width * 0.5
+         visible: lunar.festival.length > 0 && dateMatchingPrecision >= Calendar.MatchYearMonthAndDay
+    }
+
+    Timer {
+        id: showTimer
+
+        readonly property bool day: dateMatchingPrecision >= Calendar.MatchYearMonthAndDay ? true : false
+
+        interval: 1000
+        running: (day && dayStyle.containsMouse && !calendarGrid.tip.visible)
+        onTriggered: {
+            var column = index % 7
+            var alignment = column < 5 ? column : 4
+            calendarGrid.mouseOnIndex = index
+            calendarGrid.tip.text = lunar.text
+            calendarGrid.tip.show()
+            calendarGrid.tip.x = alignment * (root.borderWidth + daysCalendar.cellWidth)
+            calendarGrid.tip.y = y
+        }
+    }
+
+    Timer {
+        id: hideTimer
+        interval: 100 // ms before the tip is hidden
+        running: index == calendarGrid.mouseOnIndex && !dayStyle.containsMouse && calendarGrid.tip.visible
+        onTriggered: calendarGrid.tip.hide(); // this is the js code that hides the tip.
+                                              // you could also use visible=false; if you
+                                              // don't need animations
+    }
 }
diff -Nur plasma-framework-5.16.0/src/declarativeimports/calendar/qml/DaysCalendar.qml plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/DaysCalendar.qml
--- plasma-framework-5.16.0/src/declarativeimports/calendar/qml/DaysCalendar.qml	2015-11-08 20:40:46.000000000 +0800
+++ plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/DaysCalendar.qml	2016-01-08 09:51:21.361671636 +0800
@@ -290,6 +290,11 @@
         property bool containsTodoItems: false // FIXME
 
         property QtObject selectedDate: root.date
+
+        property alias tip: tip
+        property alias text: tip.text
+        property int mouseOnIndex: -1
+
         onSelectedDateChanged: {
             // clear the selection if the root.date is null
             if (calendarGrid.selectedDate == null) {
@@ -336,5 +341,11 @@
                 }
             }
         }
+
+        ToolTip {
+            id: tip
+            color: "#dd5b5b5b"
+            textColor: theme.backgroundColor
+        }
     }
 }
diff -Nur plasma-framework-5.16.0/src/declarativeimports/calendar/qml/ToolTip.qml plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/ToolTip.qml
--- plasma-framework-5.16.0/src/declarativeimports/calendar/qml/ToolTip.qml	1970-01-01 08:00:00.000000000 +0800
+++ plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/ToolTip.qml	2016-01-08 09:51:22.920671636 +0800
@@ -0,0 +1,51 @@
+import QtQuick 2.0
+
+Rectangle {
+    id:tooltip
+    property alias textColor: textContainer.color
+    property alias text: textContainer.text
+    property int verticalPadding: 4
+    property int horizontalPadding: 5
+    width: textContainer.width + horizontalPadding * 2
+    height: textContainer.height + verticalPadding * 2
+    color: "#aa999999"
+    
+    Text {
+        anchors.centerIn: parent
+        id:textContainer
+        text: "Gering geding ding ding!"
+        color: "black"
+    }
+    
+    NumberAnimation {
+        id: fadein
+        target: tooltip
+        property: "opacity"
+        easing.type: Easing.InOutQuad
+        duration: 300
+        from: 0
+        to: 1
+    }
+    
+    NumberAnimation {
+        id: fadeout
+        target: tooltip
+        property: "opacity"
+        easing.type: Easing.InOutQuad
+        from: 1
+        to: 0
+        onStopped: visible = false
+    }
+    
+    visible:false
+    onVisibleChanged: if(visible) fadein.start();
+    
+    function show(){
+        visible = true;
+        fadein.start();
+    }
+    
+    function hide(){
+        fadeout.start();
+    }
+} 
