diff -Nur plasma-framework-5.16.0/po/zh_CN/libplasma5.po plasma-framework-5.16.0n/po/zh_CN/libplasma5.po
--- plasma-framework-5.16.0/po/zh_CN/libplasma5.po	2016-01-08 12:29:19.202670505 +0800
+++ plasma-framework-5.16.0n/po/zh_CN/libplasma5.po	2016-01-08 12:28:33.566670511 +0800
@@ -949,3 +949,7 @@
 msgctxt "%1 is the name of the containment"
 msgid "%1 Options"
 msgstr "%1的选项"
+
+#: declarativeimports/calendar/lunar.cpp:90
+msgid "Lunar: %(NIAN)(Y)%(YUE)(M)%(RI)(D)"
+msgstr "农历: %(NIAN)年%(YUE)月%(RI)"
diff -Nur plasma-framework-5.16.0/po/zh_TW/libplasma5.po plasma-framework-5.16.0n/po/zh_TW/libplasma5.po
--- plasma-framework-5.16.0/po/zh_TW/libplasma5.po	2016-01-08 12:29:19.202670505 +0800
+++ plasma-framework-5.16.0n/po/zh_TW/libplasma5.po	2016-01-08 12:28:33.564670511 +0800
@@ -949,3 +949,7 @@
 msgctxt "%1 is the name of the containment"
 msgid "%1 Options"
 msgstr "%1 選項"
+
+#: declarativeimports/calendar/lunar.cpp:90
+msgid "Lunar: %(NIAN)(Y)%(YUE)(M)%(RI)(D)"
+msgstr "農歷: %(NIAN)年%(YUE)月%(RI)"
diff -Nur plasma-framework-5.16.0/src/declarativeimports/calendar/calendarplugin.cpp plasma-framework-5.16.0n/src/declarativeimports/calendar/calendarplugin.cpp
--- plasma-framework-5.16.0/src/declarativeimports/calendar/calendarplugin.cpp	2016-01-08 12:29:19.204670505 +0800
+++ plasma-framework-5.16.0n/src/declarativeimports/calendar/calendarplugin.cpp	2016-01-08 12:28:33.577670511 +0800
@@ -20,6 +20,8 @@
 #include "calendarplugin.h"
 #include "calendardata.h"
 #include "calendar.h"
+#include "lunar.h"
+
 #include "eventdatadecorator.h"
 #include "eventpluginsmanager.h"
 
@@ -42,6 +44,7 @@
     Q_ASSERT(uri == QLatin1String("org.kde.plasma.calendar"));
     qmlRegisterType<CalendarData>(uri, 2, 0, "CalendarData");
     qmlRegisterType<Calendar>(uri, 2, 0, "Calendar");
+    qmlRegisterType<Lunar>(uri, 2, 0, "Lunar");
     qmlRegisterType<QAbstractItemModel>();
     qmlRegisterType<QAbstractListModel>();
     qmlRegisterSingletonType<EventPluginsManager>(uri, 2, 0, "EventPluginsManager", event_plugins_manager_provider);
diff -Nur plasma-framework-5.16.0/src/declarativeimports/calendar/CMakeLists.txt plasma-framework-5.16.0n/src/declarativeimports/calendar/CMakeLists.txt
--- plasma-framework-5.16.0/src/declarativeimports/calendar/CMakeLists.txt	2016-01-08 12:29:19.206670505 +0800
+++ plasma-framework-5.16.0n/src/declarativeimports/calendar/CMakeLists.txt	2016-01-08 12:28:33.577670511 +0800
@@ -9,8 +9,13 @@
     daysmodel.cpp
     eventdatadecorator.cpp
     eventpluginsmanager.cpp
+    lunar.cpp
 )
 
+pkg_search_module(LUNARDATA REQUIRED lunar-date-2.0)
+
+include_directories(${LUNARDATA_INCLUDE_DIRS})
+
 add_library(calendarplugin SHARED ${calendar_SRCS})
 
 target_link_libraries(calendarplugin
@@ -20,6 +25,7 @@
     Qt5::Gui
     KF5::I18n
     KF5::CalendarEvents
+    ${LUNARDATA_LIBRARIES}
 )
 
 install(TARGETS calendarplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/calendar)
diff -Nur plasma-framework-5.16.0/src/declarativeimports/calendar/lunar.cpp plasma-framework-5.16.0n/src/declarativeimports/calendar/lunar.cpp
--- plasma-framework-5.16.0/src/declarativeimports/calendar/lunar.cpp	1970-01-01 08:00:00.000000000 +0800
+++ plasma-framework-5.16.0n/src/declarativeimports/calendar/lunar.cpp	2016-01-08 12:28:33.577670511 +0800
@@ -0,0 +1,96 @@
+#include <lunar-date/lunar-date.h>
+#include <stdio.h>
+#include <locale.h>
+#include <lunar.h>
+#include <QVariant>
+#include <klocalizedstring.h>
+
+Lunar::Lunar(QObject *parent)
+    : QObject(parent), m_year(-1), m_month(-1), m_day(-1)
+{}
+
+Lunar::~Lunar()
+{}
+
+void Lunar::setYear(int y)
+{
+    m_year = y;
+    emit yearChanged();
+    get();
+    m_text = "okokok";
+}
+
+void Lunar::setMonth(int m)
+{
+    m_month = m + 1;
+    emit monthChanged();
+    get();
+}
+
+void Lunar::setDay(int d)
+{
+    m_day = d;
+    emit dayChanged();
+    get();
+}
+
+void Lunar::setFestival(const QString &fest)
+{
+    m_festival = fest;
+    emit festivalChanged();
+}
+
+void Lunar::setText(const QString &text)
+{
+    m_text = text;
+    emit textChanged();
+}
+
+QString Lunar::festival() const
+{
+    return m_festival;
+}
+
+QString Lunar::text() const
+{
+    return m_text;
+}
+
+int Lunar::year() const
+{
+    return m_year;
+}
+
+int Lunar::month() const
+{
+    return m_month;
+}
+
+int Lunar::day() const
+{
+    return m_day;
+}
+
+void Lunar::get()
+{
+    if (m_year < 0 || m_month < 0 || m_day < 0)
+    {
+        m_festival.clear();
+        m_text.clear();
+        return;
+    }
+
+    GError *err = NULL;
+    LunarDate *date = lunar_date_new();
+    lunar_date_set_solar_date(date, m_year, (GDateMonth)m_month, m_day, 0, &err);
+
+    m_festival = lunar_date_get_jieri(date, "\n");
+    QString format = i18n("Lunar: %(NIAN)(Y)%(YUE)(M)%(RI)(D)");
+    format += "\n" + m_festival;
+    m_text = lunar_date_strftime(date, format.toStdString().c_str());
+
+    emit festivalChanged();
+    emit textChanged();
+
+    lunar_date_free(date);
+}
diff -Nur plasma-framework-5.16.0/src/declarativeimports/calendar/lunar.h plasma-framework-5.16.0n/src/declarativeimports/calendar/lunar.h
--- plasma-framework-5.16.0/src/declarativeimports/calendar/lunar.h	1970-01-01 08:00:00.000000000 +0800
+++ plasma-framework-5.16.0n/src/declarativeimports/calendar/lunar.h	2016-01-08 12:28:33.577670511 +0800
@@ -0,0 +1,49 @@
+#ifndef LUNAR_H
+#define LUNAR_H
+
+#include <QObject>
+#include <QString>
+
+class Lunar : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(int year READ year WRITE setYear NOTIFY yearChanged)
+    Q_PROPERTY(int month READ month WRITE setMonth NOTIFY monthChanged)
+    Q_PROPERTY(int day READ day WRITE setDay NOTIFY dayChanged)
+    Q_PROPERTY(QString festival READ festival WRITE setFestival NOTIFY festivalChanged)
+    Q_PROPERTY(QString text READ text WRITE setText NOTIFY textChanged)
+
+public:
+    Lunar(QObject *parent = 0);
+    ~Lunar();
+
+    void setYear(int y);
+    void setMonth(int m);
+    void setDay(int d);
+    void setFestival(const QString &fest);
+    void setText(const QString &text);
+
+    int year() const;
+    int month() const;
+    int day() const;
+    QString festival() const;
+    QString text() const;
+
+Q_SIGNALS:
+    void yearChanged();
+    void monthChanged();
+    void dayChanged();
+    void festivalChanged();
+    void textChanged();
+
+private:
+    void get();
+
+private:
+    QString m_festival;
+    QString m_text;
+    int m_year;
+    int m_month;
+    int m_day;
+};
+#endif
diff -Nur plasma-framework-5.16.0/src/declarativeimports/calendar/qml/DayDelegate.qml plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/DayDelegate.qml
--- plasma-framework-5.16.0/src/declarativeimports/calendar/qml/DayDelegate.qml	2016-01-08 12:29:19.209670505 +0800
+++ plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/DayDelegate.qml	2016-01-08 12:32:17.328670484 +0800
@@ -125,4 +125,53 @@
             root.date = model
         }
     }
+
+    Lunar {
+        id: lunar
+        year: thisDate.getFullYear()
+        month: thisDate.getMonth()
+        day: model.label || dayNumber
+    }
+
+    Rectangle {
+         width: 5
+         height: width
+         x: 5
+         y: 5
+         color: theme.highlightColor
+         border.color: theme.highlightColor
+         border.width: 1
+         radius: width * 0.5
+         visible: lunar.festival.length > 0 && dateMatchingPrecision >= Calendar.MatchYearMonthAndDay
+    }
+
+    Timer {
+        id: showTimer
+
+        readonly property bool day: dateMatchingPrecision >= Calendar.MatchYearMonthAndDay ? true : false
+
+        interval: 1000
+        running: (day && dayStyle.containsMouse && !calendarGrid.tip.visible)
+        onTriggered: {
+            var column = index % 7
+            var cellwidth = root.borderWidth + daysCalendar.cellWidth
+            calendarGrid.mouseOnIndex = index
+            calendarGrid.tip.text = lunar.text
+            var tipOverCell = Math.round(tip.width / cellwidth)
+            var boundCell = 7 - tipOverCell
+            var alignment = column < boundCell ? column : boundCell
+            calendarGrid.tip.show()
+            calendarGrid.tip.x = alignment * cellwidth
+            calendarGrid.tip.y = y
+        }
+    }
+
+    Timer {
+        id: hideTimer
+        interval: 100 // ms before the tip is hidden
+        running: index == calendarGrid.mouseOnIndex && !dayStyle.containsMouse && calendarGrid.tip.visible
+        onTriggered: calendarGrid.tip.hide(); // this is the js code that hides the tip.
+                                              // you could also use visible=false; if you
+                                              // don't need animations
+    }
 }
diff -Nur plasma-framework-5.16.0/src/declarativeimports/calendar/qml/DaysCalendar.qml plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/DaysCalendar.qml
--- plasma-framework-5.16.0/src/declarativeimports/calendar/qml/DaysCalendar.qml	2015-11-08 20:40:46.000000000 +0800
+++ plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/DaysCalendar.qml	2016-01-08 12:28:33.578670511 +0800
@@ -290,6 +290,11 @@
         property bool containsTodoItems: false // FIXME
 
         property QtObject selectedDate: root.date
+
+        property alias tip: tip
+        property alias text: tip.text
+        property int mouseOnIndex: -1
+
         onSelectedDateChanged: {
             // clear the selection if the root.date is null
             if (calendarGrid.selectedDate == null) {
@@ -336,5 +341,11 @@
                 }
             }
         }
+
+        ToolTip {
+            id: tip
+            color: "#dd5b5b5b"
+            textColor: theme.backgroundColor
+        }
     }
 }
diff -Nur plasma-framework-5.16.0/src/declarativeimports/calendar/qml/ToolTip.qml plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/ToolTip.qml
--- plasma-framework-5.16.0/src/declarativeimports/calendar/qml/ToolTip.qml	1970-01-01 08:00:00.000000000 +0800
+++ plasma-framework-5.16.0n/src/declarativeimports/calendar/qml/ToolTip.qml	2016-01-08 12:28:33.578670511 +0800
@@ -0,0 +1,51 @@
+import QtQuick 2.0
+
+Rectangle {
+    id:tooltip
+    property alias textColor: textContainer.color
+    property alias text: textContainer.text
+    property int verticalPadding: 4
+    property int horizontalPadding: 5
+    width: textContainer.width + horizontalPadding * 2
+    height: textContainer.height + verticalPadding * 2
+    color: "#aa999999"
+    
+    Text {
+        anchors.centerIn: parent
+        id:textContainer
+        text: "Gering geding ding ding!"
+        color: "black"
+    }
+    
+    NumberAnimation {
+        id: fadein
+        target: tooltip
+        property: "opacity"
+        easing.type: Easing.InOutQuad
+        duration: 300
+        from: 0
+        to: 1
+    }
+    
+    NumberAnimation {
+        id: fadeout
+        target: tooltip
+        property: "opacity"
+        easing.type: Easing.InOutQuad
+        from: 1
+        to: 0
+        onStopped: visible = false
+    }
+    
+    visible:false
+    onVisibleChanged: if(visible) fadein.start();
+    
+    function show(){
+        visible = true;
+        fadein.start();
+    }
+    
+    function hide(){
+        fadeout.start();
+    }
+} 
