diff --git a/src/scriptengines/qml/plasmoid/containmentinterface.cpp b/src/scriptengines/qml/plasmoid/containmentinterface.cpp
index 772031f..a8280fa 100644
--- a/src/scriptengines/qml/plasmoid/containmentinterface.cpp
+++ b/src/scriptengines/qml/plasmoid/containmentinterface.cpp
@@ -610,7 +610,6 @@ void ContainmentInterface::mimeTypeRetrieved(KIO::Job *job, const QString &mimet
         }
 
         if (!appletList.isEmpty() || !wallpaperList.isEmpty()) {
-            choices->clear();
             QHash<QAction *, QString> actionsToApplets;
             choices->addSection(i18n("Widgets"));
             foreach (const KPluginInfo &info, appletList) {
@@ -648,6 +647,8 @@ void ContainmentInterface::mimeTypeRetrieved(KIO::Job *job, const QString &mimet
                 }
             }
 
+            choices->removeAction(choices->actions().first());
+
             QAction *choice = choices->exec();
             if (choice) {
                 // Put the job on hold so it can be recycled to fetch the actual content,
