diff -Nur ark-15.12.0/app/ark_addtoservicemenu.desktop ark-15.12.0fix/app/ark_addtoservicemenu.desktop
--- ark-15.12.0/app/ark_addtoservicemenu.desktop	2015-12-15 15:42:30.000000000 +0800
+++ ark-15.12.0fix/app/ark_addtoservicemenu.desktop	2015-12-17 10:02:25.104355433 +0800
@@ -2,7 +2,7 @@
 Type=Service
 ServiceTypes=KonqPopupMenu/Plugin
 MimeType=all/all;
-Actions=compressHereAsTar;compressHereAsZip;compressHereAsRar;compressTo;
+Actions=compressHereAsTar;compressHereAsZip;compressTo;
 X-KDE-Submenu=Compress
 X-KDE-Submenu[ar]=اضغط
 X-KDE-Submenu[ast]=Comprimir
@@ -134,41 +134,6 @@
 Icon=ark
 Exec=ark --changetofirstpath --add --autofilename zip %F
 
-[Desktop Action compressHereAsRar]
-Name=Here (as RAR)
-Name[ast]=Equí (como RAR)
-Name[ca]=Aquí (com a RAR)
-Name[ca@valencia]=Ací (com a RAR)
-Name[cs]=Zde (jako RAR)
-Name[da]=Her (som RAR)
-Name[de]=Hier (Als RAR-Archiv)
-Name[en_GB]=Here (as RAR)
-Name[es]=Aquí (como RAR)
-Name[fi]=Tähän (RAR-paketiksi)
-Name[fr]=Ici (en rar)
-Name[gl]=Aquí (como RAR)
-Name[hu]=Ide (rar-ként)
-Name[it]=Qui (come RAR)
-Name[ko]=여기(RAR로)
-Name[nl]=Hier (als RAR)
-Name[pl]=Tutaj (jako RAR)
-Name[pt]=Aqui (como RAR)
-Name[pt_BR]=Aqui (como RAR)
-Name[sk]=Sem (ako RAR)
-Name[sl]=Sem (kot RAR)
-Name[sr]=овде (као РАР)
-Name[sr@ijekavian]=овде (као РАР)
-Name[sr@ijekavianlatin]=ovde (kao RAR)
-Name[sr@latin]=ovde (kao RAR)
-Name[sv]=Här (som RAR)
-Name[tr]=Buraya (RAR olarak)
-Name[uk]=Сюди (як RAR)
-Name[x-test]=xxHere (as RAR)xx
-Name[zh_CN]=此处 (作为 RAR)
-Name[zh_TW]=在此壓成 RAR 壓縮檔
-Icon=ark
-Exec=ark --changetofirstpath --add --autofilename rar %F
-
 [Desktop Action compressTo]
 Name=Compress to...
 Name[ast]=Comprimir en...
