diff -Nur ark-15.12.0/app/ark_servicemenu.desktop.cmake ark-15.12.0fix/app/ark_servicemenu.desktop.cmake
--- ark-15.12.0/app/ark_servicemenu.desktop.cmake	2015-12-15 15:42:30.000000000 +0800
+++ ark-15.12.0fix/app/ark_servicemenu.desktop.cmake	2015-12-17 10:04:54.033363535 +0800
@@ -85,6 +85,7 @@
 Name[sv]=Packa upp arkiv här
 Name[uk]=Видобути архів сюди
 Name[x-test]=xxExtract archive herexx
+Name[zh_CN]=在此解压
 Name[zh_TW]=在此解壓縮
 Icon=ark
 Exec=ark --batch --autodestination %F
@@ -117,6 +118,7 @@
 Name[sv]=Packa upp arkiv i...
 Name[uk]=Видобути архів до…
 Name[x-test]=xxExtract archive to...xx
+Name[zh_CN]=解压到...
 Name[zh_TW]=解壓縮到...
 Icon=ark
 Exec=ark --batch --autodestination --dialog %F
@@ -149,6 +151,7 @@
 Name[sv]=Packa upp arkiv här, detektera underkatalog automatiskt
 Name[uk]=Видобути архів сюди, автоматично визначити підтеку
 Name[x-test]=xxExtract archive here, autodetect subfolderxx
+Name[zh_CN]=在此解压，自动探测子文件夹
 Name[zh_TW]=在此解壓縮，自動偵測子資料夾
 Icon=ark
 Exec=ark --batch --autodestination --autosubfolder %F
