diff -Nur kio-5.16.0/src/filewidgets/knewfilemenu.cpp kio-5.16.0n/src/filewidgets/knewfilemenu.cpp
--- kio-5.16.0/src/filewidgets/knewfilemenu.cpp	2015-11-08 20:26:59.000000000 +0800
+++ kio-5.16.0n/src/filewidgets/knewfilemenu.cpp	2015-11-14 22:16:28.484198425 +0800
@@ -344,7 +344,6 @@
     KActionCollection *m_actionCollection;
     QDialog *m_fileDialog;
 
-    KActionMenu *m_menuDev;
     int m_menuItemsVersion;
     bool m_modal;
     QAction *m_newDirAction;
@@ -609,7 +608,6 @@
 {
     QMenu *menu = q->menu();
     menu->clear();
-    m_menuDev->menu()->clear();
     m_newDirAction = 0;
 
     QSet<QString> seenTexts;
@@ -703,7 +701,7 @@
                     } else if (KDesktopFile::isDesktopFile(templatePath)) {
                         KDesktopFile df(templatePath);
                         if (df.readType() == QLatin1String("FSDevice")) {
-                            m_menuDev->menu()->addAction(act);
+                            //m_menuDev->menu()->addAction(act);
                         } else {
                             menu->addAction(act);
                         }
@@ -734,8 +732,6 @@
         if (linkApp) {
             menu->addAction(linkApp);
         }
-        Q_ASSERT(m_menuDev);
-        menu->addAction(m_menuDev);
     }
 }
 
@@ -1042,8 +1038,6 @@
     if (d->m_actionCollection) {
         d->m_actionCollection->addAction(name, this);
     }
-
-    d->m_menuDev = new KActionMenu(QIcon::fromTheme(QStringLiteral("drive-removable-media")), i18n("Link to Device"), this);
 }
 
 KNewFileMenu::~KNewFileMenu()
