diff -Nur kio-5.15.0/src/filewidgets/knewfilemenu.cpp kio-5.15.0n/src/filewidgets/knewfilemenu.cpp
--- kio-5.15.0/src/filewidgets/knewfilemenu.cpp	2015-10-05 16:11:41.000000000 +0800
+++ kio-5.15.0n/src/filewidgets/knewfilemenu.cpp	2015-10-14 18:57:46.224633173 +0800
@@ -344,7 +344,6 @@
     KActionCollection *m_actionCollection;
     QDialog *m_fileDialog;
 
-    KActionMenu *m_menuDev;
     int m_menuItemsVersion;
     bool m_modal;
     QAction *m_newDirAction;
@@ -601,7 +600,6 @@
 {
     QMenu *menu = q->menu();
     menu->clear();
-    m_menuDev->menu()->clear();
     m_newDirAction = 0;
 
     QSet<QString> seenTexts;
@@ -695,7 +693,7 @@
                     } else if (KDesktopFile::isDesktopFile(templatePath)) {
                         KDesktopFile df(templatePath);
                         if (df.readType() == "FSDevice") {
-                            m_menuDev->menu()->addAction(act);
+                            //m_menuDev->menu()->addAction(act);
                         } else {
                             menu->addAction(act);
                         }
@@ -726,8 +724,6 @@
         if (linkApp) {
             menu->addAction(linkApp);
         }
-        Q_ASSERT(m_menuDev);
-        menu->addAction(m_menuDev);
     }
 }
 
@@ -1033,7 +1029,6 @@
         d->m_actionCollection->addAction(name, this);
     }
 
-    d->m_menuDev = new KActionMenu(QIcon::fromTheme("drive-removable-media"), i18n("Link to Device"), this);
 }
 
 KNewFileMenu::~KNewFileMenu()
