diff --git a/src/kioexec/CMakeLists.txt b/src/kioexec/CMakeLists.txt
index 91284a3..06b5396 100644
--- a/src/kioexec/CMakeLists.txt
+++ b/src/kioexec/CMakeLists.txt
@@ -10,6 +10,7 @@ target_link_libraries(kioexec
     KF5::KIOWidgets    # KRun
     KF5::WidgetsAddons # KMessageBox
     KF5::WindowSystem  # KStartupInfo
+    KF5::DBusAddons    # KDBusService
  )
 
 if (HAVE_X11)
diff --git a/src/kioexec/main.cpp b/src/kioexec/main.cpp
index 68ed374..6f4540f 100644
--- a/src/kioexec/main.cpp
+++ b/src/kioexec/main.cpp
@@ -37,6 +37,7 @@
 #include <KService>
 #include <KLocalizedString>
 #include <KAboutData>
+#include <KDBusService>
 
 #include <QCommandLineParser>
 #include <QCommandLineOption>
@@ -282,6 +283,7 @@ int main( int argc, char **argv )
     aboutData.addAuthor(i18n("Waldo Bastian"),QString(), QStringLiteral("bastian@kde.org"));
     aboutData.addAuthor(i18n("Oswald Buddenhagen"),QString(), QStringLiteral("ossi@kde.org"));
     KAboutData::setApplicationData(aboutData);
+    KDBusService service(KDBusService::Multiple);
 
     QCommandLineParser parser;
     parser.addHelpOption();
