diff -Nur kio-extras-15.08.3/thumbnail/imagethumbnail.desktop kio-extras-15.08.3new/thumbnail/imagethumbnail.desktop
--- kio-extras-15.08.3/thumbnail/imagethumbnail.desktop	2015-11-05 03:59:15.000000000 +0800
+++ kio-extras-15.08.3new/thumbnail/imagethumbnail.desktop	2015-11-11 15:38:38.224445781 +0800
@@ -70,6 +70,6 @@
 Name[zh_CN]=图像 (GIF、PNG、BMP ...)
 Name[zh_TW]=影像（GIF、PNG、BMP...）
 X-KDE-ServiceTypes=ThumbCreator
-MimeType=image/cgm;image/fax-g3;image/gif;image/jp2;image/png;image/tiff;image/bmp;image/x-dds;image/x-ico;image/x-jng;image/x-pcx;image/x-photo-cd;image/x-portable-bitmap;image/x-portable-graymap;image/x-portable-pixmap;image/x-rgb;image/x-tga;image/x-xbitmap;image/x-xcf;image/x-xpixmap;image/x-sun-raster;image/vnd.adobe.photoshop;image/x-psd;image/x-hdr;image/x-pic;image/vnd.microsoft.icon;image/x-icon;image/x-webp;image/webp;
+MimeType=image/cgm;image/fax-g3;image/gif;image/jp2;image/png;image/tiff;image/bmp;image/x-dds;image/x-ico;image/x-jng;image/x-pcx;image/x-photo-cd;image/x-portable-bitmap;image/x-portable-graymap;image/x-portable-pixmap;image/x-rgb;image/x-tga;image/x-xbitmap;image/x-xcf;image/x-xpixmap;image/x-sun-raster;image/vnd.adobe.photoshop;image/x-psd;image/x-hdr;image/x-pic;image/vnd.microsoft.icon;image/x-icon;image/x-webp;image/webp;image/x-icns;
 X-KDE-Library=imagethumbnail
 CacheThumbnail=true
