diff -Nur dolphin-15.11.80/src/panels/places/placesitemmodel.cpp dolphin-15.11.80n/src/panels/places/placesitemmodel.cpp
--- dolphin-15.11.80/src/panels/places/placesitemmodel.cpp	2015-11-18 07:39:00.000000000 +0800
+++ dolphin-15.11.80n/src/panels/places/placesitemmodel.cpp	2015-11-21 16:32:02.514054273 +0800
@@ -1161,7 +1161,6 @@
 QUrl PlacesItemModel::searchUrlForType(const QString& type)
 {
     Baloo::Query query;
-    query.addType("File");
     query.addType(type);
 
     return query.toSearchUrl();
