diff -Naur kxmlgui-5.16.0.bak/src/kmainwindow.cpp kxmlgui-5.16.0/src/kmainwindow.cpp
--- kxmlgui-5.16.0.bak/src/kmainwindow.cpp	2015-12-08 10:14:29.598885711 +0800
+++ kxmlgui-5.16.0/src/kmainwindow.cpp	2015-12-09 15:47:32.229562710 +0800
@@ -179,11 +179,6 @@
 
 void KMainWindowPrivate::init(KMainWindow *_q)
 {
-    // We set allow quit to true when the first mainwindow is created, so that when the refcounting
-    // reaches 0 the application can quit. We don't want this to happen before the first mainwindow
-    // is created, otherwise running a job in main would exit the app too early.
-    QCoreApplication::setQuitLockEnabled(true);
-
     q = _q;
 
     q->setAnimated(q->style()->styleHint(QStyle::SH_Widget_Animate, 0, q));
