diff --git a/src/kstatusnotifieritem.cpp b/src/kstatusnotifieritem.cpp
index f9bf460..2fa8f55 100644
--- a/src/kstatusnotifieritem.cpp
+++ b/src/kstatusnotifieritem.cpp
@@ -665,7 +665,7 @@ const int KStatusNotifierItemPrivate::s_protocolVersion = 0;
 KStatusNotifierItemPrivate::KStatusNotifierItemPrivate(KStatusNotifierItem *item)
     : q(item),
       category(KStatusNotifierItem::ApplicationStatus),
-      status(KStatusNotifierItem::Passive),
+      status(KStatusNotifierItem::Active),
       movie(0),
       menu(0),
       titleAction(0),
