--- a/mdb.c	2014-12-11 16:47:11.000000000 +0800
+++ b/mdb.c	2015-11-04 14:39:30.103505326 +0800
@@ -2605,6 +2605,10 @@
 	MDB_ntxn *ntxn;
 	int rc, size, tsize = sizeof(MDB_txn);
 
+    if (env == NULL) {
+        DPUTS("environment had fatal error, must shutdown!");
+        return MDB_PANIC;
+    }
 	if (env->me_flags & MDB_FATAL_ERROR) {
 		DPUTS("environment had fatal error, must shutdown!");
 		return MDB_PANIC;
