# The s390 architecture needs a pause before accessing the test DB.

diff -up lmdb-LMDB_0.9.16/libraries/liblmdb/Makefile.old lmdb-LMDB_0.9.16/libraries/liblmdb/Makefile
--- lmdb-LMDB_0.9.16/libraries/liblmdb/Makefile.old	2015-08-14 11:08:02.226631788 +0200
+++ lmdb-LMDB_0.9.16/libraries/liblmdb/Makefile	2015-08-14 11:08:58.618906240 +0200
@@ -51,7 +51,7 @@ clean:
 
 test:	all
 	rm -rf testdb && mkdir testdb
-	./mtest && ./mdb_stat testdb
+	./mtest && sleep 1 && ./mdb_stat testdb
 
 liblmdb.a:	mdb.o midl.o
 	ar rs $@ mdb.o midl.o
