diff --git a/Modules/about-distro/src/Module.cpp b/Modules/about-distro/src/Module.cpp
index bec3a36..85653a8 100644
--- a/Modules/about-distro/src/Module.cpp
+++ b/Modules/about-distro/src/Module.cpp
@@ -101,7 +101,7 @@ void Module::load()
     QString logoPath = cg.readEntry("LogoPath", QString());
     QPixmap logo;
     if (logoPath.isEmpty())
-        logo = QIcon::fromTheme(QStringLiteral("start-here-kde")).pixmap(128, 128);
+        logo = QIcon::fromTheme(QStringLiteral("isoft-logo"), QIcon::fromTheme(QStringLiteral("start-here-kde"))).pixmap(128, 128);
     else
         logo = QPixmap(logoPath);
     ui->logoLabel->setPixmap(logo);
