diff --git a/Modules/about-distro/src/Module.cpp b/Modules/about-distro/src/Module.cpp
index fa8d09d..6eb4a15 100644
--- a/Modules/about-distro/src/Module.cpp
+++ b/Modules/about-distro/src/Module.cpp
@@ -23,6 +23,7 @@
 
 #include <QIcon>
 #include <QStandardPaths>
+#include <QFile>
 
 #include <kcoreaddons_version.h>
 #include <KAboutData>
@@ -128,7 +129,8 @@ void Module::load()
     QString logoPath = cg.readEntry("LogoPath", QString());
     QPixmap logo;
     if (logoPath.isEmpty()) {
-        logo = QIcon::fromTheme(QStringLiteral("start-here-kde")).pixmap(128, 128);
+        logoPath = "/usr/share/isoft-artwork/logo/128x128/isoft-logo.png";
+        logo = QFile::exists(logoPath) ? QPixmap(logoPath) : QIcon::fromTheme("start-here-kde").pixmap(128, 128);
     } else {
         logo = QPixmap(logoPath);
     }
