diff -Nur user-manager-5.2.90+git20150330/src/accountinfo.cpp user-manager-5.2.90+git20150330n/src/accountinfo.cpp
--- user-manager-5.2.90+git20150330/src/accountinfo.cpp	2015-03-30 17:48:24.000000000 +0000
+++ user-manager-5.2.90+git20150330n/src/accountinfo.cpp	2015-07-10 11:42:27.579860202 +0000
@@ -170,9 +170,15 @@
         }
     }
     if (m_infoToSave.contains(AccountModel::Face)) {
+        //the faceicon of KDE works as:
+        //kuser_unix.cpp in kcoreaddons hardcode it to ~/.face.icon
+        //kickoff menu in plasma desktop use this path.
+        //we better follow the kde settings.
+        //But it's still need more works.
+        //for example, change kickoff menu face icon automatically.
         const QString path = m_infoToSave[AccountModel::Face].toString();
         QString faceFile = QDesktopServices::storageLocation(QDesktopServices::HomeLocation);
-        faceFile.append(QLatin1String("/.face"));
+        faceFile.append(QLatin1String("/.face.icon"));
 
         QFile::remove(faceFile);
         KIO::CopyJob* moveJob = KIO::move(KUrl(path), KUrl(faceFile), KIO::HideProgressInfo);
