diff -Nur sddm-0.11.0/src/greeter/UserModel.cpp sddm-0.11.0n/src/greeter/UserModel.cpp
--- sddm-0.11.0/src/greeter/UserModel.cpp	2014-11-20 15:47:24.000000000 +0000
+++ sddm-0.11.0n/src/greeter/UserModel.cpp	2015-07-10 11:38:38.674864301 +0000
@@ -77,9 +77,15 @@
             user->gid = int(current_pw->pw_gid);
 
             // search for face icon
+            // here we first search accountsservice user icons.
+            // The better way is to use dbus to query the path, But we do not want this behaviour too heavy.
+            // So, let's just hardcode.
+            QString accountsServiceFace = QString("/var/lib/AccountsService/icons/%1").arg(user->name);
             QString userFace = QString("%1/.face.icon").arg(user->homeDir);
             QString systemFace = QString("%1/%2.face.icon").arg(mainConfig.Theme.FacesDir.get()).arg(user->name);
-            if (QFile::exists(userFace))
+            if (QFile::exists(accountsServiceFace))
+                user->icon = accountsServiceFace;
+            else if (QFile::exists(userFace))
                 user->icon = userFace;
             else if (QFile::exists(systemFace))
                 user->icon = systemFace;
