diff -Nur kscreen-5.4.3/CMakeLists.txt kscreen-5.4.3n/CMakeLists.txt
--- kscreen-5.4.3/CMakeLists.txt	2015-11-05 20:38:14.000000000 +0800
+++ kscreen-5.4.3n/CMakeLists.txt	2015-11-07 17:01:19.396548798 +0800
@@ -30,6 +30,7 @@
 add_subdirectory(kded)
 add_subdirectory(tests)
 add_subdirectory(console)
+add_subdirectory(osd)
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
diff -Nur kscreen-5.4.3/kcm/src/kcm_kscreen.cpp kscreen-5.4.3n/kcm/src/kcm_kscreen.cpp
--- kscreen-5.4.3/kcm/src/kcm_kscreen.cpp	2015-11-05 20:37:32.000000000 +0800
+++ kscreen-5.4.3n/kcm/src/kcm_kscreen.cpp	2015-11-07 16:59:23.634550871 +0800
@@ -85,6 +85,7 @@
     }
 
     mKScreenWidget->setConfig(qobject_cast<GetConfigOperation*>(op)->config());
+    mKScreenWidget->getShowOsd();
 }
 
 KCMKScreen::~KCMKScreen()
@@ -99,6 +100,8 @@
         return;
     }
 
+    mKScreenWidget->setShowOsd();
+
     const KScreen::ConfigPtr &config = mKScreenWidget->currentConfig();
 
     bool atLeastOneEnabledOutput = false;
diff -Nur kscreen-5.4.3/kcm/src/scalingconfig.cpp kscreen-5.4.3n/kcm/src/scalingconfig.cpp
--- kscreen-5.4.3/kcm/src/scalingconfig.cpp	2015-11-05 20:37:32.000000000 +0800
+++ kscreen-5.4.3n/kcm/src/scalingconfig.cpp	2015-11-07 16:59:23.634550871 +0800
@@ -46,7 +46,7 @@
 
     ui.previewWidget->setScale(1);
     ui.scaleLabel->setText(QString::number(1));
-    
+
     load();
 }
 
diff -Nur kscreen-5.4.3/kcm/src/widget.cpp kscreen-5.4.3n/kcm/src/widget.cpp
--- kscreen-5.4.3/kcm/src/widget.cpp	2015-11-05 20:37:32.000000000 +0800
+++ kscreen-5.4.3n/kcm/src/widget.cpp	2015-11-07 16:59:23.635550871 +0800
@@ -1,5 +1,6 @@
 /*
- * Copyright (C) 2013  Daniel Vr??til <dvratil@redhat.com>
+ * Copyright (C) 2013  Daniel Vrátil <dvratil@redhat.com>
+ * Copyright (C) 2015 Leslie Zhai <xiang.zhai@i-soft.com.cn>
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -24,6 +25,9 @@
 #endif
 #include "primaryoutputcombo.h"
 
+#include <QCheckBox>
+#include <QSettings>
+#include <QMessageBox>
 #include <QVBoxLayout>
 #include <QSplitter>
 #include <QLabel>
@@ -55,7 +59,8 @@
     QWidget(parent),
     mScreen(0),
     mConfig(0),
-    mPrevConfig(0)
+    mPrevConfig(0),
+    mShowOsd(nullptr)
 {
     qRegisterMetaType<QQuickView*>();
 
@@ -124,9 +129,14 @@
                 dialog->exec();
                 delete dialog;
             });
-    
+
     vbox->addWidget(setScaleButton);
 
+    mShowOsd = new QCheckBox(i18n("Automatically popup switch wizard when new monitor plugged in"));
+    getShowOsd();
+    connect(mShowOsd, &QCheckBox::clicked, this, &Widget::slotShowOsdChanged);
+    vbox->addWidget(mShowOsd);
+
     mOutputTimer = new QTimer(this);
     connect(mOutputTimer, &QTimer::timeout,
             this, &Widget::clearOutputIdentifiers);
@@ -139,6 +149,64 @@
     clearOutputIdentifiers();
 }
 
+void Widget::slotShowOsdChanged() 
+{
+    QMessageBox msgBox(
+        QMessageBox::Question,
+        i18n("Hide osd when new monitor plugged in?"),
+        i18n("You have disabled automatically popping up of monitor switch wizard, it can be triggered by Multimedia hotkey 'XF86Display'. If you have no hotkey supported, consider to setup the hotkey or keep the option unchanged"),
+        QMessageBox::Cancel | QMessageBox::Ok);
+
+    msgBox.setDefaultButton(QMessageBox::Cancel);
+
+    int state = mShowOsd->checkState();
+    if (state == Qt::Unchecked) {
+        int ret = msgBox.exec();
+        if (ret == QMessageBox::Ok) {
+            mShowOsd->setChecked(false);
+            emit changed();
+        } else {
+            mShowOsd->setChecked(true);
+        }
+    } else if (state == Qt::Checked) {
+        emit changed();
+    }
+}
+
+void Widget::getShowOsd() 
+{
+    mShowOsd->setChecked(isShowOsd());
+}
+
+void Widget::setShowOsd() 
+{
+    QSettings settings("kscreen", "settings");
+    int state = mShowOsd->checkState();
+    if (state == Qt::Unchecked)
+        settings.setValue("osd/showme", false);
+    else if (state == Qt::Checked)
+        settings.setValue("osd/showme", true);
+}
+
+bool Widget::isShowOsd()
+{
+    QSettings settings("kscreen", "settings");
+
+    QString settingsDir = QDir::homePath() + "/.config/kscreen";
+    QDir dir(settingsDir);
+    if (!dir.exists()) {
+        dir.mkdir(settingsDir);
+        return true;
+    }
+
+    QString settingsPath = settingsDir + "/settings.conf";
+    QFile file(settingsPath);
+    if (!file.exists())
+        return true;
+
+    return settings.value("osd/showme").toBool();
+}
+
 bool Widget::eventFilter(QObject* object, QEvent* event)
 {
     if (event->type() == QEvent::Resize) {
diff -Nur kscreen-5.4.3/kcm/src/widget.h kscreen-5.4.3n/kcm/src/widget.h
--- kscreen-5.4.3/kcm/src/widget.h	2015-11-05 20:37:32.000000000 +0800
+++ kscreen-5.4.3n/kcm/src/widget.h	2015-11-07 16:59:23.635550871 +0800
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2013  Daniel Vrátil <dvratil@redhat.com>
+ * Copyright (C) 2015 Leslie Zhai <xiang.zhai@i-soft.com.cn>
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -33,6 +34,7 @@
 
 class QPushButton;
 class QComboBox;
+class QCheckBox;
 
 class QQuickView;
 
@@ -52,6 +54,9 @@
     void setConfig(const KScreen::ConfigPtr &config);
     KScreen::ConfigPtr currentConfig() const;
 
+    void getShowOsd();
+    void setShowOsd();
+
   protected:
     virtual bool eventFilter(QObject *object, QEvent *event);
 
@@ -73,11 +78,14 @@
     void slotIdentifyOutputs(KScreen::ConfigOperation *op);
     void clearOutputIdentifiers();
 
+    void slotShowOsdChanged();
+
   private:
     void loadQml();
     void initPrimaryCombo();
 
     KScreen::OutputPtr findOutput(const KScreen::ConfigPtr &config, const QVariantMap &info);
+    bool isShowOsd();
 
   private:
     QMLScreen *mScreen;
@@ -97,6 +105,7 @@
     QList<QQuickView*> mOutputIdentifiers;
     QTimer *mOutputTimer;
 
+    QCheckBox *mShowOsd;
 };
 
 #endif // WIDGET_H
diff -Nur kscreen-5.4.3/kded/CMakeLists.txt kscreen-5.4.3n/kded/CMakeLists.txt
--- kscreen-5.4.3/kded/CMakeLists.txt	2015-11-05 20:37:32.000000000 +0800
+++ kscreen-5.4.3n/kded/CMakeLists.txt	2015-11-07 16:59:23.635550871 +0800
@@ -21,7 +21,7 @@
 
 add_library(kded_kscreen MODULE ${kscreen_daemon_SRCS})
 
-target_link_libraries(kded_kscreen Qt5::Widgets Qt5::DBus KF5::Screen KF5::DBusAddons KF5::I18n KF5::XmlGui KF5::GlobalAccel)
+target_link_libraries(kded_kscreen Qt5::DBus KF5::Screen KF5::DBusAddons KF5::I18n KF5::XmlGui KF5::GlobalAccel KF5::Service)
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/kscreen.desktop.cmake
                ${CMAKE_CURRENT_BINARY_DIR}/kscreen.desktop
diff -Nur kscreen-5.4.3/kded/daemon.cpp kscreen-5.4.3n/kded/daemon.cpp
--- kscreen-5.4.3/kded/daemon.cpp	2015-11-05 20:37:32.000000000 +0800
+++ kscreen-5.4.3n/kded/daemon.cpp	2015-11-07 16:59:23.636550871 +0800
@@ -1,5 +1,6 @@
 /*************************************************************************************
  *  Copyright (C) 2012 by Alejandro Fiestas Olivares <afiestas@kde.org>              *
+ *  Copyright (C) 2015 Leslie Zhai <xiang.zhai@i-soft.com.cn>                        *
  *                                                                                   *
  *  This program is free software; you can redistribute it and/or                    *
  *  modify it under the terms of the GNU General Public License                      *
@@ -32,6 +33,7 @@
 #include <KActionCollection>
 #include <KPluginFactory>
 #include <KGlobalAccel>
+#include <KToolInvocation>
 
 #include <kscreen/config.h>
 #include <kscreen/output.h>
@@ -217,14 +219,8 @@
 
 void KScreenDaemon::applyGenericConfig()
 {
-    if (m_iteration == Generator::ExtendToRight) {
-        m_iteration = Generator::None;
-    }
-
-    m_iteration = Generator::DisplaySwitchAction(static_cast<int>(m_iteration) + 1);
-    qCDebug(KSCREEN_KDED) << "displayButton: " << m_iteration;
-
-    doApplyConfig(Generator::self()->displaySwitch(m_iteration));
+    // TODO: disable display switch, but just popup osd
+    KToolInvocation::kdeinitExec(QString("kscreen-osd"), QStringList());
 }
 
 void KScreenDaemon::lidClosedChanged(bool lidIsClosed)
@@ -307,6 +303,8 @@
             Q_EMIT unknownOutputConnected(output->name());
         }
     }
+
+    KToolInvocation::kdeinitExec(QString("kscreen-osd"), QStringList());
 }
 
 
diff -Nur kscreen-5.4.3/kded/generator.cpp kscreen-5.4.3n/kded/generator.cpp
--- kscreen-5.4.3/kded/generator.cpp	2015-11-05 20:37:32.000000000 +0800
+++ kscreen-5.4.3n/kded/generator.cpp	2015-11-07 16:59:23.637550871 +0800
@@ -193,25 +193,6 @@
     }
 
     switch (action) {
-    case Generator::ExtendToLeft: {
-        qCDebug(KSCREEN_KDED) << "Extend to left";
-        external->setPos(QPoint(0,0));
-        external->setEnabled(true);
-        const KScreen::ModePtr extMode = bestModeForOutput(external);
-        Q_ASSERT(extMode);
-        external->setCurrentModeId(extMode->id());
-
-        Q_ASSERT(external->currentMode()); // we must have a mode now
-        const QSize size = external->currentMode()->size();
-        embedded->setPos(QPoint(size.width(), 0));
-        embedded->setEnabled(true);
-        embedded->setPrimary(true);
-        const KScreen::ModePtr embeddedMode = bestModeForOutput(embedded);
-        Q_ASSERT(embeddedMode);
-        embedded->setCurrentModeId(embeddedMode->id());
-
-        return config;
-    }
     case Generator::TurnOffEmbedded: {
         qCDebug(KSCREEN_KDED) << "Turn off embedded (laptop)";
         embedded->setEnabled(false);
diff -Nur kscreen-5.4.3/kded/generator.h kscreen-5.4.3n/kded/generator.h
--- kscreen-5.4.3/kded/generator.h	2015-11-05 20:37:32.000000000 +0800
+++ kscreen-5.4.3n/kded/generator.h	2015-11-07 16:59:23.638550871 +0800
@@ -1,5 +1,6 @@
 /*************************************************************************************
  *  Copyright (C) 2012 by Alejandro Fiestas Olivares <afiestas@kde.org>              *
+ *  Copyright (C) 2015 Leslie Zhai <xiang.zhai@i-soft.com.cn>                        *
  *                                                                                   *
  *  This program is free software; you can redistribute it and/or                    *
  *  modify it under the terms of the GNU General Public License                      *
@@ -36,10 +37,9 @@
         enum DisplaySwitchAction {
             None = 0,
             Clone = 1,
-            ExtendToLeft = 2,
-            TurnOffEmbedded = 3,
-            TurnOffExternal = 4,
-            ExtendToRight = 5,
+            TurnOffEmbedded = 2,
+            TurnOffExternal = 3,
+            ExtendToRight = 4,
         };
 
         static Generator* self();
diff -Nur kscreen-5.4.3/CMakeLists.txt kscreen-5.4.3new/CMakeLists.txt
--- kscreen-5.4.3/CMakeLists.txt	2015-11-07 17:14:44.073534388 +0800
+++ kscreen-5.4.3new/CMakeLists.txt	2015-11-07 17:14:58.490534130 +0800
@@ -11,7 +11,7 @@
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 
 find_package(Qt5 REQUIRED COMPONENTS QuickWidgets Test)
-find_package(KF5 REQUIRED COMPONENTS DBusAddons Config ConfigWidgets I18n XmlGui GlobalAccel WidgetsAddons)
+find_package(KF5 REQUIRED COMPONENTS DBusAddons Config ConfigWidgets I18n XmlGui GlobalAccel WidgetsAddons Service)
 
 set(MIN_LIBKSCREEN_VERSION "5.2.91")
 find_package(KF5Screen ${MIN_LIBKSCREEN_VERSION} REQUIRED)
