diff -Naur kdeclarative-5.25.0/src/kpackagelauncherqml/main.cpp kdeclarative-master/src/kpackagelauncherqml/main.cpp
--- kdeclarative-5.25.0/src/kpackagelauncherqml/main.cpp	2016-08-07 21:44:43.000000000 +0800
+++ kdeclarative-master/src/kpackagelauncherqml/main.cpp	2016-08-29 12:08:40.688374301 +0800
@@ -92,6 +92,9 @@
     for (auto author : data.authors()) {
         aboutData.addAuthor(author.name(), author.task(), author.emailAddress(), author.webAddress(), author.ocsUsername());
     }
+    // We assume that desktop file in applicaitons dir is named similar as
+    // the pluginId, this is required to get icons working on wayland for example
+    aboutData.setDesktopFileName(data.pluginId());
 
     KAboutData::setApplicationData(aboutData);
 
diff -Naur kdeclarative-5.25.0/src/qmlcontrols/kquickcontrolsaddons/icondialog.cpp kdeclarative-master/src/qmlcontrols/kquickcontrolsaddons/icondialog.cpp
--- kdeclarative-5.25.0/src/qmlcontrols/kquickcontrolsaddons/icondialog.cpp	2016-08-07 21:44:43.000000000 +0800
+++ kdeclarative-master/src/qmlcontrols/kquickcontrolsaddons/icondialog.cpp	2016-08-29 12:08:40.715374302 +0800
@@ -22,30 +22,36 @@
 #include <QQuickItem>
 #include <QQuickWindow>
 
+#include <QApplication>
 #include <KIconDialog>
 #include <KIconLoader>
 
 IconDialog::IconDialog(QObject *parent)
     : QObject(parent)
-    , m_dialog(new KIconDialog())
+    , m_dialog(Q_NULLPTR)
     , m_iconSize(0)
     , m_user(false)
     , m_modality(Qt::WindowModal)
     , m_visible(false)
 {
-    connect(m_dialog.data(), &KIconDialog::newIconName, this, [this](const QString &newIconName) {
-        if (m_iconName != newIconName) {
-            m_iconName = newIconName;
-            emit iconNameChanged(newIconName);
-        }
-    });
+    if (qobject_cast<QApplication *>(QCoreApplication::instance())) {
+        m_dialog.reset(new KIconDialog());
+        connect(m_dialog.data(), &KIconDialog::newIconName, this, [this](const QString &newIconName) {
+                if (m_iconName != newIconName) {
+                m_iconName = newIconName;
+                emit iconNameChanged(newIconName);
+                }
+        });
 
-    m_dialog->installEventFilter(this);
+        m_dialog->installEventFilter(this);
+    }
 }
 
 IconDialog::~IconDialog()
 {
-    m_dialog->close();
+    if (m_dialog) {
+        m_dialog->close();
+    }
 }
 
 QString IconDialog::iconName() const
diff -Naur kdeclarative-5.25.0/src/quickaddons/CMakeLists.txt kdeclarative-master/src/quickaddons/CMakeLists.txt
--- kdeclarative-5.25.0/src/quickaddons/CMakeLists.txt	2016-08-07 21:44:43.000000000 +0800
+++ kdeclarative-master/src/quickaddons/CMakeLists.txt	2016-08-29 12:08:40.719374302 +0800
@@ -1,8 +1,13 @@
-add_library(KF5QuickAddons
+set(KF5QuickAddons_LIB_SRCS
             imagetexturescache.cpp
             managedtexturenode.cpp
             quickviewsharedengine.cpp
-            configmodule.cpp)
+            configmodule.cpp
+            qtquicksettings.cpp)
+kconfig_add_kcfg_files(KF5QuickAddons_LIB_SRCS renderersettings.kcfgc)
+
+add_library(KF5QuickAddons ${KF5QuickAddons_LIB_SRCS})
+
 generate_export_header(KF5QuickAddons EXPORT_FILE_NAME ${CMAKE_CURRENT_BINARY_DIR}/quickaddons_export.h BASE_NAME QuickAddons)
 add_library(KF5::QuickAddons ALIAS KF5QuickAddons)
 
@@ -15,6 +20,7 @@
   PRIVATE
     KF5::I18n
     KF5::Declarative
+    KF5::ConfigGui
 )
 
 set_target_properties(KF5QuickAddons
@@ -29,7 +35,6 @@
   HEADER_NAMES
   ImageTexturesCache
   ManagedTextureNode
-
   PREFIX QuickAddons
   REQUIRED_HEADERS QuickAddons_HEADERS
 )
@@ -40,6 +45,7 @@
   HEADER_NAMES
   ImageTexturesCache
   ManagedTextureNode
+  QtQuickSettings
   ConfigModule
   QuickViewSharedEngine
 
diff -Naur kdeclarative-5.25.0/src/quickaddons/qtquicksettings.cpp kdeclarative-master/src/quickaddons/qtquicksettings.cpp
--- kdeclarative-5.25.0/src/quickaddons/qtquicksettings.cpp	1970-01-01 08:00:00.000000000 +0800
+++ kdeclarative-master/src/quickaddons/qtquicksettings.cpp	2016-08-29 12:08:40.720374302 +0800
@@ -0,0 +1,43 @@
+/*
+ *   Copyright 2016 David Edmundson <davidedmundson@kde.org>
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU Library General Public License as
+ *   published by the Free Software Foundation; either version 2 or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details
+ *
+ *   You should have received a copy of the GNU Library General Public
+ *   License along with this program; if not, write to the
+ *   Free Software Foundation, Inc.,
+ *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include "qtquicksettings.h"
+#include "renderersettings.h"
+
+#include <QSurfaceFormat>
+
+void KQuickAddons::QtQuickSettings::init()
+{
+    PlasmaQtQuickSettings::RendererSettings s;
+    if (!s.renderLoop().isNull()) {
+        qputenv("QSG_RENDER_LOOP", s.renderLoop().toLatin1());
+    }
+
+    if (!s.sceneGraphBackend().isNull()) {
+        qputenv("QMLSCENE_DEVICE", s.sceneGraphBackend().toLatin1());
+        //TODO Qt5.8 - QQuickWindow::setSceneGraphBackend(..)
+    }
+    if (s.forceGlCoreProfile()) {
+        QSurfaceFormat format;
+        format.setVersion(3,2);
+        format.setProfile(QSurfaceFormat::CoreProfile);
+        QSurfaceFormat::setDefaultFormat(format);
+    }
+}
+
diff -Naur kdeclarative-5.25.0/src/quickaddons/qtquicksettings.h kdeclarative-master/src/quickaddons/qtquicksettings.h
--- kdeclarative-5.25.0/src/quickaddons/qtquicksettings.h	1970-01-01 08:00:00.000000000 +0800
+++ kdeclarative-master/src/quickaddons/qtquicksettings.h	2016-08-29 12:08:40.726374302 +0800
@@ -0,0 +1,42 @@
+/*
+ *   Copyright 2016 David Edmundson <davidedmundson@kde.org>
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU Library General Public License as
+ *   published by the Free Software Foundation; either version 2 or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details
+ *
+ *   You should have received a copy of the GNU Library General Public
+ *   License along with this program; if not, write to the
+ *   Free Software Foundation, Inc.,
+ *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#ifndef PLASMAQTQUICKSETTINGS_H
+#define PLASMAQTQUICKSETTINGS_H
+
+#include "quickaddons_export.h"
+
+namespace KQuickAddons
+{
+    namespace QtQuickSettings
+    {
+    /*
+     * Load settings from the appConfig (app + kdeglobals) which relate to QtQuick rendering.
+     * This will then set relevant default QWindow and QSurfaceFormat flags, or in some cases environment variables
+     *
+     * This function must be called at the start of your application before any windows are created.
+     *
+     * @since 5.26
+     */
+    QUICKADDONS_EXPORT void init();
+
+    }
+}
+
+#endif
diff -Naur kdeclarative-5.25.0/src/quickaddons/quickviewsharedengine.cpp kdeclarative-master/src/quickaddons/quickviewsharedengine.cpp
--- kdeclarative-5.25.0/src/quickaddons/quickviewsharedengine.cpp	2016-08-07 21:44:43.000000000 +0800
+++ kdeclarative-master/src/quickaddons/quickviewsharedengine.cpp	2016-08-29 12:08:40.726374302 +0800
@@ -156,6 +156,16 @@
 }
 
 
+void QuickViewSharedEngine::setTranslationDomain(const QString &translationDomain)
+{
+    d->qmlObject->setTranslationDomain(translationDomain);
+}
+
+QString QuickViewSharedEngine::translationDomain() const
+{
+    return d->qmlObject->translationDomain();
+}
+
 QQmlEngine *QuickViewSharedEngine::engine() const
 {
     return d->qmlObject->engine();
diff -Naur kdeclarative-5.25.0/src/quickaddons/quickviewsharedengine.h kdeclarative-master/src/quickaddons/quickviewsharedengine.h
--- kdeclarative-5.25.0/src/quickaddons/quickviewsharedengine.h	2016-08-07 21:44:43.000000000 +0800
+++ kdeclarative-master/src/quickaddons/quickviewsharedengine.h	2016-08-29 12:08:40.727374302 +0800
@@ -56,6 +56,29 @@
     explicit QuickViewSharedEngine(QWindow *parent = 0);
     ~QuickViewSharedEngine();
 
+    /**
+     * Installs a translation domain for all
+     * i18n global functions. If a translation domain is set all i18n calls delegate to the
+     * matching i18nd calls with the provided translation domain.
+     *
+     * The translationDomain affects all i18n calls including those from imports. Because of
+     * that modules intended to be used as imports should prefer the i18nd variants and set
+     * the translation domain explicitly in each call.
+     *
+     * This method is only required if your declarative usage is inside a library. If it's
+     * in an application there is no need to set the translation domain as the application's
+     * domain can be used.
+     *
+     * @param translationDomain The translation domain to be used for i18n calls.
+     * @since 5.25
+     */
+    void setTranslationDomain(const QString &translationDomain);
+
+    /**
+     * @return the translation domain for the i18n calls done in this QML engine
+     * @since 5.25
+     */
+    QString translationDomain() const;
 
     QQmlEngine *engine() const;
     QList<QQmlError> errors() const;
diff -Naur kdeclarative-5.25.0/src/quickaddons/renderer.kcfg kdeclarative-master/src/quickaddons/renderer.kcfg
--- kdeclarative-5.25.0/src/quickaddons/renderer.kcfg	1970-01-01 08:00:00.000000000 +0800
+++ kdeclarative-master/src/quickaddons/renderer.kcfg	2016-08-29 12:08:40.727374302 +0800
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<kcfg xmlns="http://www.kde.org/standards/kcfg/1.0"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xsi:schemaLocation="http://www.kde.org/standards/kcfg/1.0
+                          http://www.kde.org/standards/kcfg/1.0/kcfg.xsd" >
+    <kcfgfile arg="true" />
+    <group name="QtQuickRendererSettings">
+        <entry name="RenderLoop" type="String">
+            <default></default>
+        </entry>
+        <entry name="SceneGraphBackend" type="String">
+            <default>false</default>
+        </entry>
+        <entry name="ForceGlCoreProfile" type="Bool">
+            <default>false</default>
+        </entry>
+    </group>
+</kcfg>
diff -Naur kdeclarative-5.25.0/src/quickaddons/renderersettings.kcfgc kdeclarative-master/src/quickaddons/renderersettings.kcfgc
--- kdeclarative-5.25.0/src/quickaddons/renderersettings.kcfgc	1970-01-01 08:00:00.000000000 +0800
+++ kdeclarative-master/src/quickaddons/renderersettings.kcfgc	2016-08-29 12:08:40.727374302 +0800
@@ -0,0 +1,4 @@
+File=renderer.kcfg
+NameSpace=PlasmaQtQuickSettings
+ClassName=RendererSettings
+UseEnumTypes=true
