diff -Naur kde-cli-tools-5.4.3.old/keditfiletype/filegroupdetails.cpp kde-cli-tools-5.4.3/keditfiletype/filegroupdetails.cpp
--- kde-cli-tools-5.4.3.old/keditfiletype/filegroupdetails.cpp	2015-12-01 09:14:39.249518991 +0800
+++ kde-cli-tools-5.4.3/keditfiletype/filegroupdetails.cpp	2015-12-01 10:52:52.863675969 +0800
@@ -24,7 +24,7 @@
 #include <QRadioButton>
 #include <QButtonGroup>
 #include <QGroupBox>
-
+#include <QFile>
 #include <klocalizedstring.h>
 
 FileGroupDetails::FileGroupDetails(QWidget *parent)
@@ -34,7 +34,10 @@
 
   QGroupBox *autoEmbedBox = new QGroupBox( i18n("Left Click Action (only for Konqueror file manager)") );
   m_autoEmbed = new QButtonGroup( autoEmbedBox );
+  QFile file("/usr/bin/konqueror");
+  if(file.exists()){
   secondLayout->addWidget( autoEmbedBox );
+    }
   // The order of those two items is very important. If you change it, fix typeslistitem.cpp !
   QRadioButton *r1 = new QRadioButton( i18n("Show file in embedded viewer"));
   QRadioButton *r2 = new QRadioButton( i18n("Show file in separate viewer"));
diff -Naur kde-cli-tools-5.4.3.old/keditfiletype/filetypedetails.cpp kde-cli-tools-5.4.3/keditfiletype/filetypedetails.cpp
--- kde-cli-tools-5.4.3.old/keditfiletype/filetypedetails.cpp	2015-12-01 09:14:39.249518991 +0800
+++ kde-cli-tools-5.4.3/keditfiletype/filetypedetails.cpp	2015-12-01 10:51:18.079675980 +0800
@@ -31,6 +31,7 @@
 #include <QListWidget>
 #include <QPushButton>
 #include <QRadioButton>
+#include <QFile>
 
 // KDE
 #include <kconfig.h>
@@ -139,8 +140,10 @@
   QVBoxLayout *secondLayout = new QVBoxLayout(secondWidget);
 
   m_autoEmbedBox = new QGroupBox( i18n("Left Click Action in Konqueror"), secondWidget );
-  secondLayout->addWidget( m_autoEmbedBox );
-
+  QFile file("/usr/bin/konqueror");
+  if(file.exists()){
+    secondLayout->addWidget( m_autoEmbedBox );
+  }
   m_autoEmbedBox->setSizePolicy( QSizePolicy::Preferred, QSizePolicy::Fixed );
 
   QRadioButton *embViewerRadio = new QRadioButton( i18n("Show file in embedded viewer") );
@@ -170,12 +173,17 @@
     " always shows files in a separate viewer.") );
 
   embedServiceListWidget = new KServiceListWidget( KServiceListWidget::SERVICELIST_SERVICES, secondWidget );
-//  embedServiceListWidget->setMinimumHeight( serviceListWidget->sizeHint().height() );
+  if(file.exists()){
+  embedServiceListWidget->setMinimumHeight( serviceListWidget->sizeHint().height() );
+  }
   connect( embedServiceListWidget, SIGNAL(changed(bool)), this, SIGNAL(changed(bool)));
+  if(file.exists()){
   secondLayout->addWidget(embedServiceListWidget);
-
+  }
   m_tabWidget->addTab( firstWidget, i18n("&General") );
+  if(file.exists()){
   m_tabWidget->addTab( secondWidget, i18n("&Embedding") );
+  } 
 }
 
 void FileTypeDetails::updateRemoveButton()
