diff --git a/app/kded.h b/app/kded.h
index 68bc6a3..0393a3a 100644
--- a/app/kded.h
+++ b/app/kded.h
@@ -59,6 +59,11 @@ class KHotKeysModule
          */
         Q_SCRIPTABLE QString get_menuentry_shortcut(const QString &storageId);
 
+        /**
+         * declare daemon settings outdated, do not write back until re-read has been triggered
+         */
+        Q_SCRIPTABLE Q_NOREPLY void declareConfigOutdated();
+
     private Q_SLOTS:
 
         //! Save
diff --git a/app/kded.cpp b/app/kded.cpp
index 34eafea..1a70481 100644
--- a/app/kded.cpp
+++ b/app/kded.cpp
@@ -45,7 +45,6 @@ KHotKeysModule::KHotKeysModule(QObject* parent, const QList<QVariant>&)
     {
     // initialize
     qDebug() << "Installing the delayed initialization callback.";
-    Settings::thisIsTheDaemon = true;
     QMetaObject::invokeMethod( this, "initialize", Qt::QueuedConnection);
     }
 
@@ -258,4 +257,9 @@ void KHotKeysModule::save()
     KHotKeys::khotkeys_set_active( true );
     }
 
+void KHotKeysModule::declareConfigOutdated()
+    {
+    Settings::isOutdated = true;
+    }
+
 #include "kded.moc"
diff --git a/kcm_hotkeys/kcm_hotkeys.cpp b/kcm_hotkeys/kcm_hotkeys.cpp
index c57d9c5..b37ef1b 100644
--- a/kcm_hotkeys/kcm_hotkeys.cpp
+++ b/kcm_hotkeys/kcm_hotkeys.cpp
@@ -345,15 +345,10 @@ bool KCMHotkeysPrivate::maybeShowWidget(const QModelIndex &nextIndex)
 
 void KCMHotkeysPrivate::save()
     {
-    if (current)
-        applyCurrentItem();
-
-    // Write the settings
-    model->save();
-
+    bool daemonFailed = false;
     if (!KHotKeys::Daemon::isRunning())
         {
-        if (!KHotKeys::Daemon::start())
+        if (KHotKeys::Daemon::start())
             {
             // On startup the demon does the updating stuff, therefore reload
             // the actions.
@@ -361,12 +356,8 @@ void KCMHotkeysPrivate::save()
             }
         else
             {
-            KMessageBox::error(
-                q,
-                "<qt>" + i18n("Unable to contact khotkeys. Your changes are saved, but they could not be activated.") + "</qt>" );
+            daemonFailed = true;
             }
-
-        return;
         }
 
     // Inform kdedkhotkeys demon to reload settings
@@ -385,14 +376,34 @@ void KCMHotkeysPrivate::save()
             {
             qCritical() << err.name() << ":" << err.message();
             }
-        KMessageBox::error(
-            q,
-            "<qt>" + i18n("Unable to contact khotkeys. Your changes are saved, but they could not be activated.") + "</qt>" );
-        return;
+        daemonFailed = true;
+        }
+
+    if (!daemonFailed)
+        iface->declareConfigOutdated();
+
+    if (current)
+        applyCurrentItem(); // mutex on
+
+    // Write the settings
+    model->save();
+
+    if(!iface->isValid())
+        {
+        err = iface->lastError();
+        if (err.isValid())
+            {
+            qCritical() << err.name() << ":" << err.message();
+            }
+        daemonFailed = true;
         }
 
     // Reread the configuration. We have no possibility to check if it worked.
-    iface->reread_configuration();
+    if (daemonFailed)
+        KMessageBox::error(q, "<qt>" +
+                           i18n("Unable to contact khotkeys. Your changes are saved, but they could not be activated.") + "</qt>" );
+    else
+        iface->reread_configuration(); // mutex off
     }
 
 
diff --git a/libkhotkeysprivate/settings.h b/libkhotkeysprivate/settings.h
index b12622f..6e36c00 100644
--- a/libkhotkeysprivate/settings.h
+++ b/libkhotkeysprivate/settings.h
@@ -181,7 +181,7 @@ public:
      */
     bool isConfigFileValid(KConfigBase const &config, ImportType ask);
 
-    static bool thisIsTheDaemon;
+    static bool isOutdated;
 
 protected:
 
diff --git a/libkhotkeysprivate/settings.cpp b/libkhotkeysprivate/settings.cpp
index 6d44173..a944c72 100644
--- a/libkhotkeysprivate/settings.cpp
+++ b/libkhotkeysprivate/settings.cpp
@@ -36,7 +36,7 @@ namespace KHotKeys
 {
 
 // Settings
-bool Settings::thisIsTheDaemon = false;
+bool Settings::isOutdated = false;
 
 Settings::Settings()
     : m_actions( NULL ),
@@ -497,8 +497,7 @@ bool Settings::read_settings(ActionDataGroup *root, KConfigBase const &config, b
             return false;
         }
 
-    if (thisIsTheDaemon)
-        mainGroup.deleteEntry("ConfigUpdated");
+    Settings::isOutdated = false;
     return true;
     }
 
@@ -529,16 +528,11 @@ bool Settings::update()
 
 void Settings::write()
     {
-    KConfig cfg( KHOTKEYS_CONFIG_FILE );
-    if (thisIsTheDaemon && KConfigGroup(&cfg, "Main").readEntry("ConfigUpdated", false))
+    if (isOutdated)
         return; // the kcm wrote after we read it and we're trying to write because
                 // we got an update from kglobalaccel - ie. it's pointless and our
                 // info is dated. The kcm should next ask us to read settings
-    if (!thisIsTheDaemon)
-        {
-        KConfigGroup(&cfg, "Main").writeEntry("ConfigUpdated", true);
-        cfg.sync();
-        }
+    KConfig cfg( KHOTKEYS_CONFIG_FILE );
     SettingsWriter writer(this, Retain);
     writer.writeTo(cfg);
     }
