diff -Nur powerdevil-5.4.3-clean/kcmodule/activities/activitywidget.cpp powerdevil-5.4.3/kcmodule/activities/activitywidget.cpp
--- powerdevil-5.4.3-clean/kcmodule/activities/activitywidget.cpp	2015-12-15 22:53:14.138563518 -0600
+++ powerdevil-5.4.3/kcmodule/activities/activitywidget.cpp	2015-12-16 02:04:00.829636484 -0600
@@ -73,6 +73,8 @@
     connect(m_ui->actLikeComboBox, SIGNAL(currentIndexChanged(int)), this, SLOT(setChanged()));
     connect(m_ui->alwaysActionBox, SIGNAL(currentIndexChanged(int)), this, SLOT(setChanged()));
     connect(m_ui->alwaysAfterSpin, SIGNAL(valueChanged(int)), this, SLOT(setChanged()));
+    connect(m_ui->noShutdownPCBox, SIGNAL(toggled(bool)), this, SLOT(setChanged()));
+    connect(m_ui->alwaysBox, SIGNAL(toggled(bool)), this, SLOT(setChanged()));
 
     connect(m_actionEditWidget, SIGNAL(changed(bool)), this, SIGNAL(changed(bool)));
 }
@@ -91,6 +93,7 @@
 
     QSet< Solid::PowerManagement::SleepState > methods = Solid::PowerManagement::supportedSleepStates();
 
+    m_ui->alwaysActionBox->clear();
     if (methods.contains(Solid::PowerManagement::SuspendState)) {
         m_ui->alwaysActionBox->addItem(QIcon::fromTheme("system-suspend"),
                                        i18n("Sleep"), (uint)SuspendSession::ToRamMode);
@@ -183,6 +186,11 @@
 
 void ActivityWidget::setChanged()
 {
+    if (sender() == m_ui->noShutdownPCBox && m_ui->noShutdownPCBox->isChecked())
+        m_ui->alwaysBox->setChecked(false);
+    if (sender() == m_ui->alwaysBox && m_ui->alwaysBox->isChecked())
+        m_ui->noShutdownPCBox->setChecked(false);
+
     emit changed(true);
 }
 
