commit 728dba09749282cf7abcc2e0ede9a28315e83793
Author: Michal Humpula <michal.humpula@hudrydum.cz>
Date:   Sun Nov 29 11:24:42 2015 +0100

    fix opening new files

diff --git a/urlinfo.h b/urlinfo.h
index 897140a..74a8d44 100644
--- a/urlinfo.h
+++ b/urlinfo.h
@@ -33,7 +33,7 @@ struct UrlInfo
         : cursor(KTextEditor::Cursor::invalid())
     {
         // convert to an url
-        const QRegularExpression withProtocol(QStringLiteral("^[a-zA-Z]+:")); // TODO: remove after Qt supports this on its own
+        const QRegularExpression withProtocol(QStringLiteral("^[a-zA-Z]+://")); // TODO: remove after Qt supports this on its own
         if (withProtocol.match(path).hasMatch()) {
             url = QUrl::fromUserInput(path);
         } else {
@@ -50,7 +50,7 @@ struct UrlInfo
                 int line = match.captured(1).toInt() - 1;
                 // don't use an invalid column when the line is valid
                 int column = qMax(0, match.captured(2).toInt() - 1);
-                url = QUrl::fromLocalFile(path);
+                url = QUrl::fromLocalFile(QDir::current().absoluteFilePath(path));
                 cursor = {line, column};
             }
         }
