diff -Nur baloo-5.16.0/src/engine/queryparser.cpp baloo-5.16.0n/src/engine/queryparser.cpp
--- baloo-5.16.0/src/engine/queryparser.cpp	2015-11-20 16:43:12.467800734 +0800
+++ baloo-5.16.0n/src/engine/queryparser.cpp	2015-11-20 16:46:51.880796804 +0800
@@ -161,14 +161,25 @@
         queries << phraseQueries;
         phraseQueries.clear();
     }
-    //detect text contains chinese or not.
-    //if contain chinese, every chinese character should be a term.
+    //detect text contains CJKV or not.
+    //if contain CJKV, every CJKV character should be a term.
+    //according to http://stackoverflow.com/questions/1366068/whats-the-complete-range-for-chinese-characters-in-unicode
+    //Block                                   Range       Comment
+    //CJK Unified Ideographs                  4E00-9FFF   Common
+    //CJK Unified Ideographs Extension A      3400-4DFF   Rare
+    //CJK Unified Ideographs Extension B      20000-2A6DF Rare, historic
+    //CJK Compatibility Ideographs            F900-FAFF   Duplicates, unifiable variants, corporate characters
+    //CJK Compatibility Ideographs Supplement 2F800-2FA1F Unifiable variants
     int nCount = text.count();
     for(int i = 0 ; i < nCount ; i++)
     {
         QChar cha = text.at(i);
         ushort uni = cha.unicode();
-        if(uni >= 0x4E00 && uni <= 0x9FA5)
+        if((uni >= 0x4E00 && uni <= 0x9FFF) ||
+           (uni >= 0x3400 && uni <= 0x4DFF) ||
+           (uni >= 0x20000 && uni <= 0x2A6DF) ||
+           (uni >= 0xF900 && uni <= 0xFAFF) ||
+           (uni >= 0x2F800 && uni <= 0x2FA1F))
         {  
            queries << EngineQuery(QString(cha).toUtf8(), EngineQuery::StartsWith);
         }   
diff -Nur baloo-5.16.0/src/engine/termgenerator.cpp baloo-5.16.0n/src/engine/termgenerator.cpp
--- baloo-5.16.0/src/engine/termgenerator.cpp	2015-11-20 16:43:12.467800734 +0800
+++ baloo-5.16.0n/src/engine/termgenerator.cpp	2015-11-20 16:46:43.575796953 +0800
@@ -99,14 +99,25 @@
 void TermGenerator::indexFileNameText(const QString& text, const QByteArray& prefix, int wdfInc)
 {
     QStringList terms = termList(text);
-    //detect text contains chinese or not.
-    //if contain chinese, every chinese character should be a term.
+    //detect text contains CJKV or not.
+    //if contain CJKV, every CJKV character should be a term.
+    //according to http://stackoverflow.com/questions/1366068/whats-the-complete-range-for-chinese-characters-in-unicode
+    //Block                                   Range       Comment
+    //CJK Unified Ideographs                  4E00-9FFF   Common
+    //CJK Unified Ideographs Extension A      3400-4DFF   Rare
+    //CJK Unified Ideographs Extension B      20000-2A6DF Rare, historic
+    //CJK Compatibility Ideographs            F900-FAFF   Duplicates, unifiable variants, corporate characters
+    //CJK Compatibility Ideographs Supplement 2F800-2FA1F Unifiable variants
     int nCount = text.count();  
     for(int i = 0 ; i < nCount ; i++)  
     {  
         QChar cha = text.at(i);  
         ushort uni = cha.unicode();  
-        if(uni >= 0x4E00 && uni <= 0x9FA5)  
+        if((uni >= 0x4E00 && uni <= 0x9FFF) ||
+           (uni >= 0x3400 && uni <= 0x4DFF) ||
+           (uni >= 0x20000 && uni <= 0x2A6DF) ||
+           (uni >= 0xF900 && uni <= 0xFAFF) ||
+           (uni >= 0x2F800 && uni <= 0x2FA1F))
         {
            terms<<QString(cha);  
         }  
