diff -Nur oxygen-fonts-5.3.2/CMakeLists.txt oxygen-fonts-5.3.2n/CMakeLists.txt
--- oxygen-fonts-5.3.2/CMakeLists.txt	2015-06-25 11:13:56.000000000 +0000
+++ oxygen-fonts-5.3.2n/CMakeLists.txt	2015-07-06 10:20:47.838075803 +0000
@@ -35,32 +35,12 @@
   set(OXYGEN_FONT_INSTALL_DIR  ${CMAKE_INSTALL_PREFIX}/${SHARE_INSTALL_PREFIX}/fonts/truetype/oxygen)
 endif()
 
-find_program(FONTFORGE_EXECUTABLE NAMES fontforge)
-if(FONTFORGE_EXECUTABLE)
-    message(STATUS "FontForge found: ${FONTFORGE_EXECUTABLE}")
-else()
-    message(FATAL_ERROR "Could not find FontForge executable")
-endif()
-
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
-add_custom_target(regular ALL fontforge ${CMAKE_CURRENT_SOURCE_DIR}/generate-ttf.pe
-                   ${CMAKE_CURRENT_SOURCE_DIR}/oxygen-fonts/Regular-400/src/Oxygen-Sans.sfd
-                   ${CMAKE_CURRENT_BINARY_DIR}/Oxygen-Sans.ttf
-                  )
-add_custom_target(bold ALL fontforge ${CMAKE_CURRENT_SOURCE_DIR}/generate-ttf.pe
-                   ${CMAKE_CURRENT_SOURCE_DIR}/oxygen-fonts/Bold-700/src/Oxygen-Sans-Bold.sfd
-                   ${CMAKE_CURRENT_BINARY_DIR}/Oxygen-Sans-Bold.ttf
-                  )
-add_custom_target(mono ALL fontforge ${CMAKE_CURRENT_SOURCE_DIR}/generate-ttf.pe
-                   ${CMAKE_CURRENT_SOURCE_DIR}/oxygen-fonts/mono-400/src/OxygenMono-Regular.sfd
-                   ${CMAKE_CURRENT_BINARY_DIR}/OxygenMono-Regular.ttf
-                  )
-
 set(fontFiles
-    ${CMAKE_CURRENT_BINARY_DIR}/OxygenMono-Regular.ttf
-    ${CMAKE_CURRENT_BINARY_DIR}/Oxygen-Sans-Bold.ttf
-    ${CMAKE_CURRENT_BINARY_DIR}/Oxygen-Sans.ttf
+    ${CMAKE_CURRENT_SOURCE_DIR}/oxygen-fonts/mono-400/OxygenMono-Regular.ttf
+    ${CMAKE_CURRENT_SOURCE_DIR}/oxygen-fonts/Bold-700/Oxygen-Sans-Bold.ttf
+    ${CMAKE_CURRENT_SOURCE_DIR}/oxygen-fonts/Regular-400/Oxygen-Sans.ttf
 )
 
 install(FILES ${fontFiles} DESTINATION ${OXYGEN_FONT_INSTALL_DIR})
