diff --git a/applets/kicker/package/contents/ui/CompactRepresentation.qml b/applets/kicker/package/contents/ui/CompactRepresentation.qml
index 56d1c7c..769e6ae 100644
--- a/applets/kicker/package/contents/ui/CompactRepresentation.qml
+++ b/applets/kicker/package/contents/ui/CompactRepresentation.qml
@@ -21,6 +21,7 @@ import QtQuick 2.0
 import QtQuick.Layouts 1.1
 
 import org.kde.plasma.core 2.0 as PlasmaCore
+import org.kde.plasma.private.kicker 0.1 as Kicker
 
 Item {
     id: root
@@ -28,6 +29,10 @@ Item {
     property bool vertical: (plasmoid.formFactor == PlasmaCore.Types.Vertical)
     property QtObject dashWindow: null
 
+    Kicker.ProcessRunner {
+        id: processRunner
+    }
+
     PlasmaCore.IconItem {
         id: buttonIcon

@@ -35,7 +40,7 @@ Item {
 
         visible: !(plasmoid.configuration.useCustomButtonImage && plasmoid.configuration.customButtonImage)
 
-        source: plasmoid.configuration.icon
+        source: processRunner.isFileExists("/usr/share/isoft-artwork/logo/scalable/isoft-logo-symbolic.svgz") ? "/usr/share/isoft-artwork/logo/scalable/isoft-logo-symbolic.svgz" : plasmoid.configuration.icon
         active: mouseArea.containsMouse
 
         onWidthChanged: {
diff --git a/applets/kicker/plugin/processrunner.cpp b/applets/kicker/plugin/processrunner.cpp
index 9214284..140295a 100644
--- a/applets/kicker/plugin/processrunner.cpp
+++ b/applets/kicker/plugin/processrunner.cpp
@@ -20,6 +20,7 @@
 #include "processrunner.h"
 
 #include <KProcess>
+#include <QFile>
 
 ProcessRunner::ProcessRunner(QObject *parent) : QObject(parent)
 {
@@ -33,3 +34,8 @@ void ProcessRunner::runMenuEditor()
 {
     KProcess::startDetached("kmenuedit");
 }
+
+bool ProcessRunner::isFileExists(QString path)
+{
+    return QFile::exists(path);
+}
diff --git a/applets/kicker/plugin/processrunner.h b/applets/kicker/plugin/processrunner.h
index 8390bc8..0f8ab62 100644
--- a/applets/kicker/plugin/processrunner.h
+++ b/applets/kicker/plugin/processrunner.h
@@ -31,6 +31,7 @@ class ProcessRunner : public QObject
         ~ProcessRunner();
 
         Q_INVOKABLE void runMenuEditor();
+        Q_INVOKABLE bool isFileExists(QString path);
 };
 
 #endif
