diff --git a/containments/desktop/package/contents/ui/FolderView.qml b/containments/desktop/package/contents/ui/FolderView.qml
index 578ec87..6ec5784 100644
--- a/containments/desktop/package/contents/ui/FolderView.qml
+++ b/containments/desktop/package/contents/ui/FolderView.qml
@@ -670,6 +670,14 @@ Item {
                     } else if (event.key == Qt.Key_End) {
                         currentIndex = count - 1;
                         updateSelection(event.modifiers);
+                    } else if (event.modifiers && Qt.ControlModifier) {
+                        if (event.key == Qt.Key_V) {
+                            dir.paste();
+                        } else if (event.key == Qt.Key_C) {
+                            dir.copy();
+                        } else if (event.key == Qt.Key_X) {
+                            dir.cut();
+                        }
                     }
                 }
 
diff --git a/containments/desktop/plugins/folder/foldermodel.h b/containments/desktop/plugins/folder/foldermodel.h
index 5d7f886..d8ceae3 100644
--- a/containments/desktop/plugins/folder/foldermodel.h
+++ b/containments/desktop/plugins/folder/foldermodel.h
@@ -191,6 +191,10 @@ class FolderModel : public QSortFilterProxyModel
         bool isDir(const QModelIndex &index, const KDirModel *dirModel) const;
         bool lessThan(const QModelIndex &left, const QModelIndex &right) const;
 
+        Q_INVOKABLE void paste();
+        Q_INVOKABLE void copy();
+        Q_INVOKABLE void cut();
+
     Q_SIGNALS:
         void urlChanged() const;
         void resolvedUrlChanged() const;
@@ -220,9 +224,6 @@ class FolderModel : public QSortFilterProxyModel
         void statResult(KJob *job);
         void evictFromIsDirCache(const KFileItemList &items);
         void selectionChanged(QItemSelection selected, QItemSelection deselected);
-        void copy();
-        void cut();
-        void paste();
         void pasteTo();
         void refresh();
         void moveSelectedToTrash();
