diff --git a/applets/kickoff/core/leavemodel.cpp b/applets/kickoff/core/leavemodel.cpp
index fe33ae2..8cba572 100644
--- a/applets/kickoff/core/leavemodel.cpp
+++ b/applets/kickoff/core/leavemodel.cpp
@@ -185,18 +185,6 @@ void LeaveModel::updateModel()
         appendRow(standbyOption);
     }
 
-    if (spdMethods.contains(Solid::PowerManagement::SuspendState)) {
-        QStandardItem *suspendramOption = createStandardItem("leave:/suspendram");
-        suspendramOption->setData(system, Kickoff::GroupNameRole);
-        appendRow(suspendramOption);
-    }
-
-    if (spdMethods.contains(Solid::PowerManagement::HibernateState)) {
-        QStandardItem *suspenddiskOption = createStandardItem("leave:/suspenddisk");
-        suspenddiskOption->setData(system, Kickoff::GroupNameRole);
-        appendRow(suspenddiskOption);
-    }
-
     if (canLogout) {
         if (KWorkSpace::canShutDown(KWorkSpace::ShutdownConfirmDefault, KWorkSpace::ShutdownTypeReboot)) {
             // Restart
