diff --git a/containments/desktop/plugins/folder/internallibkonq/konq_popupmenu.cpp b/containments/desktop/plugins/folder/internallibkonq/konq_popupmenu.cpp
index 7745b53..6614d58 100644
--- a/containments/desktop/plugins/folder/internallibkonq/konq_popupmenu.cpp
+++ b/containments/desktop/plugins/folder/internallibkonq/konq_popupmenu.cpp
@@ -513,8 +513,15 @@ KFileItemActions* KonqPopupMenu::fileItemActions() const
 
 void KonqPopupMenuPrivate::slotPopupNewView()
 {
-    Q_FOREACH(const QUrl& url, m_popupItemProperties.urlList()) {
-        (void) new KRun(url, m_parentWidget);
+    Q_FOREACH (const KFileItem& item, m_popupItemProperties.items()) {
+        if (item.isLink()) {
+            (void) new KRun(QUrl::fromLocalFile(item.linkDest()), m_parentWidget);
+        } else {
+            if (item.isDesktopFile())
+                (void) new KRun(item.url(), m_parentWidget);
+            else
+                (void) new KRun(item.targetUrl(), m_parentWidget);
+        }
     }
 }
 
