diff --git a/containments/desktop/package/contents/ui/main.qml b/containments/desktop/package/contents/ui/main.qml
index 8c42706..cf68a07 100644
--- a/containments/desktop/package/contents/ui/main.qml
+++ b/containments/desktop/package/contents/ui/main.qml
@@ -32,7 +32,7 @@ import "LayoutManager.js" as LayoutManager
 
 DragDrop.DropArea {
     id: root
-    objectName: isFolder ? "folder" : "desktop"
+    objectName: "folder"
 
     width: {
         if (isContainment || !folderViewLayer.ready) {
@@ -60,7 +60,7 @@ DragDrop.DropArea {
         return height
     }
 
-    property bool isFolder: (plasmoid.pluginName == "org.kde.plasma.folder")
+    property bool isFolder: true
     property bool isContainment: ("containmentType" in plasmoid)
     property bool isPopup: (plasmoid.location != PlasmaCore.Types.Floating)
 
