diff --git a/applets/kickoff/package/contents/ui/FavoritesView.qml b/applets/kickoff/package/contents/ui/FavoritesView.qml
index 901b80a2343700ca4054bc46aee703f0cfd26372..5b7a2cf1e5f1607fcdc2b8a88ac714837891574a 100644
--- a/applets/kickoff/package/contents/ui/FavoritesView.qml
+++ b/applets/kickoff/package/contents/ui/FavoritesView.qml
@@ -104,6 +104,7 @@ Item {
             }
 
             targetRow = Math.min(kickoffListView.count, targetRow);
+            targetRow = Math.max(0, targetRow);
 
             kickoffListView.model.moveRow(startRow,  targetRow);
 
