diff -Nur plasma-desktop-5.3.95/layout-templates/org.kde.plasma.desktop.defaultPanel/contents/layout.js plasma-desktop-5.3.95new/layout-templates/org.kde.plasma.desktop.defaultPanel/contents/layout.js
--- plasma-desktop-5.3.95/layout-templates/org.kde.plasma.desktop.defaultPanel/contents/layout.js	2015-08-13 15:12:07.937752722 +0000
+++ plasma-desktop-5.3.95new/layout-templates/org.kde.plasma.desktop.defaultPanel/contents/layout.js	2015-08-13 15:13:05.448755322 +0000
@@ -38,3 +38,6 @@
 panel.addWidget("org.kde.plasma.systemtray")
 panel.addWidget("org.kde.plasma.digitalclock")
 
+//by default, lock the panel.
+locked=true
+
