diff -Nur plasma-desktop-5.4.2/kcms/desktoppaths/desktoppath.desktop plasma-desktop-5.4.2new/kcms/desktoppaths/desktoppath.desktop
--- plasma-desktop-5.4.2/kcms/desktoppaths/desktoppath.desktop	2015-10-01 17:51:50.000000000 +0800
+++ plasma-desktop-5.4.2new/kcms/desktoppaths/desktoppath.desktop	2015-10-14 20:56:09.049800750 +0800
@@ -7,8 +7,8 @@
 
 X-KDE-Library=kcm_desktoppaths
 X-KDE-ParentApp=kcontrol
-X-KDE-System-Settings-Parent-Category=applications
-X-KDE-Weight=60
+##X-KDE-System-Settings-Parent-Category=applications
+##X-KDE-Weight=60
 
 Name=Locations
 Name[ca]=Ubicacions
