--- plasma-desktop-5.4.2/layout-templates/org.kde.plasma.desktop.defaultPanel/contents/layout.js	2015-10-01 17:51:50.000000000 +0800
+++ plasma-desktop-5.4.2n/layout-templates/org.kde.plasma.desktop.defaultPanel/contents/layout.js	2015-10-29 17:55:51.696103424 +0800
@@ -31,9 +31,11 @@
 kickoff.currentConfigGroup = ["Shortcuts"]
 kickoff.writeConfig("global", "Alt+F1")
 
+panel.addWidget("org.kde.plasma.showdesktop")
 //panel.addWidget("org.kde.plasma.showActivityManager")
 panel.addWidget("org.kde.plasma.pager")
 panel.addWidget("org.kde.plasma.taskmanager")
+//panel.addWidget("org.kde.plasma.kimpanel")
 panel.addWidget("org.kde.plasma.systemtray")
 panel.addWidget("org.kde.plasma.digitalclock")
 
