diff -Nur plasma-desktop-5.3.2/kcms/keyboard/xkb_rules.cpp plasma-desktop-5.3.2n/kcms/keyboard/xkb_rules.cpp
--- plasma-desktop-5.3.2/kcms/keyboard/xkb_rules.cpp	2015-06-25 11:08:54.000000000 +0000
+++ plasma-desktop-5.3.2n/kcms/keyboard/xkb_rules.cpp	2015-07-21 21:43:40.326857320 +0000
@@ -157,29 +157,7 @@
 
 QString Rules::findXkbDir()
 {
-	QString xkbParentDir;
-
-	QString base(XLIBDIR);
-	if( base.count('/') >= 3 ) {
-		// .../usr/lib/X11 -> /usr/share/X11/xkb vs .../usr/X11/lib -> /usr/X11/share/X11/xkb
-		QString delta = base.endsWith("X11") ? "/../../share/X11" : "/../share/X11";
-		QDir baseDir(base + delta);
-		if( baseDir.exists() ) {
-			xkbParentDir = baseDir.absolutePath();
-		}
-		else {
-			QDir baseDir(base + "/X11");	// .../usr/X11/lib/X11/xkb (old XFree)
-			if( baseDir.exists() ) {
-				xkbParentDir = baseDir.absolutePath();
-			}
-		}
-	}
-
-	if( xkbParentDir.isEmpty() ) {
-		xkbParentDir = "/usr/share/X11";
-	}
-
-	return xkbParentDir + "/xkb";
+	return "/usr/share/X11/xkb";
 }
 
 static QString findXkbRulesFile()
