diff -Nur plasma-desktop-5.4.2nn/containments/desktop/package/contents/config/main.xml plasma-desktop-5.4.2/containments/desktop/package/contents/config/main.xml
--- plasma-desktop-5.4.2nn/containments/desktop/package/contents/config/main.xml	2015-10-14 15:07:44.124055636 +0800
+++ plasma-desktop-5.4.2/containments/desktop/package/contents/config/main.xml	2015-10-14 15:12:07.286242668 +0800
@@ -99,10 +99,10 @@
     </entry>
 
     <entry name="showTweaks" type="Bool">
-      <default>true</default>
+      <default>false</default>
     </entry>
     <entry name="showToolbox" type="Bool">
-      <default>true</default>
+      <default>false</default>
     </entry>
     <entry name="pressToMove" type="Bool">
       <default>false</default>
diff -Nur plasma-desktop-5.4.2nn/containments/desktop/package/contents/ui/ConfigLocation.qml plasma-desktop-5.4.2/containments/desktop/package/contents/ui/ConfigLocation.qml
--- plasma-desktop-5.4.2nn/containments/desktop/package/contents/ui/ConfigLocation.qml	2015-10-14 15:07:44.123055628 +0800
+++ plasma-desktop-5.4.2/containments/desktop/package/contents/ui/ConfigLocation.qml	2015-10-14 15:08:15.109313552 +0800
@@ -45,9 +45,6 @@
         if (cfg_url == "desktop:/") {
             locationDesktop.checked = true;
             locationCustomValue.text = "";
-        } else if (cfg_url == "activities:/current/") {
-            locationCurrentActivity.checked = true;
-            locationCustomValue.text = "";
         } else {
             var placeForUrl = placesModel.indexForUrl(cfg_url);
 
@@ -86,9 +83,7 @@
                     onCurrentChanged: {
                         if (current == locationDesktop) {
                             cfg_url = "desktop:/";
-                        } else if (current == locationCurrentActivity) {
-                            cfg_url = "activities:/current/";
-                        }
+                        } 
                     }
                 }
 
@@ -98,13 +93,6 @@
                     exclusiveGroup: locationGroup
                 }
 
-                RadioButton {
-                    id: locationCurrentActivity
-                    visible: placesModel.activityLinkingEnabled
-                    text: i18n("Show files linked to the current activity")
-                    exclusiveGroup: locationGroup
-                }
-
                 GridLayout {
                     RadioButton {
                         id: locationPlace
diff -Nur plasma-desktop-5.4.2nn/containments/desktop/package/contents/ui/ConfigTweaks.qml plasma-desktop-5.4.2/containments/desktop/package/contents/ui/ConfigTweaks.qml
--- plasma-desktop-5.4.2nn/containments/desktop/package/contents/ui/ConfigTweaks.qml	2015-10-14 15:07:44.123055628 +0800
+++ plasma-desktop-5.4.2/containments/desktop/package/contents/ui/ConfigTweaks.qml	2015-10-14 15:09:33.427964937 +0800
@@ -30,7 +30,6 @@
     width: childrenRect.width
     height: childrenRect.height
 
-    property alias cfg_showToolbox: showToolbox.checked
     property alias cfg_pressToMove: pressToMove.checked
 
     ColumnLayout {
@@ -45,23 +44,6 @@
         }
 
         GroupBox {
-            Layout.fillWidth: true
-
-            title: i18n("Desktop Layout")
-            flat: true
-
-            ColumnLayout {
-                Layout.fillWidth: true
-
-                CheckBox {
-                    id: showToolbox
-
-                    text: i18n("Show the desktop toolbox")
-                }
-            }
-        }
-
-        GroupBox {
             Layout.fillWidth: true
 
             title: i18n("Widget Handling")
