diff --git a/applets/kickoff/core/itemhandlers.cpp b/applets/kickoff/core/itemhandlers.cpp
index e703bb8..ed9ab2e 100644
--- a/applets/kickoff/core/itemhandlers.cpp
+++ b/applets/kickoff/core/itemhandlers.cpp
@@ -31,6 +31,7 @@
 #include <KToolInvocation>
 #include <KUrl>
 #include <Solid/PowerManagement>
+#include <KRun>
 
 // KDE Base
 #include <kworkspace.h>
@@ -57,6 +58,9 @@ bool ServiceItemHandler::openUrl(const KUrl& url)
         if (service) {
             RecentApplications::self()->add(service);
         } else {
+            if (url.toString().startsWith("file:///usr/share/applications") == false) {
+                KRun::runUrl(url, "application/x-desktop", 0);
+            }
             qWarning() << "Failed to find service for" << url;
             return false;
         }
diff --git a/applets/kickoff/package/contents/ui/RecentlyUsedView.qml b/applets/kickoff/package/contents/ui/RecentlyUsedView.qml
index df25db2..00b2452 100644
--- a/applets/kickoff/package/contents/ui/RecentlyUsedView.qml
+++ b/applets/kickoff/package/contents/ui/RecentlyUsedView.qml
@@ -45,7 +45,7 @@ BaseView {
             text: i18n("Clear Recent Documents")
             icon: "edit-clear-history"
             onClicked: recentlyUsedModel.clearRecentDocuments();
-            visible: !contextMenu.isApp
+            visible: true
         }
     }
 }
