diff -Nur plasma-desktop-5.4.2/kcms/input/CMakeLists.txt plasma-desktop-5.4.2fix/kcms/input/CMakeLists.txt
--- plasma-desktop-5.4.2/kcms/input/CMakeLists.txt	2015-10-01 17:51:50.000000000 +0800
+++ plasma-desktop-5.4.2fix/kcms/input/CMakeLists.txt	2015-10-18 10:39:31.320076342 +0800
@@ -24,44 +24,7 @@
 install(TARGETS kapplymousetheme ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 
-########### next target ###############
-
-set(kcm_input_PART_SRCS mouse.cpp main.cpp)
-
-set(klauncher_xml ${KINIT_DBUS_INTERFACES_DIR}/kf5_org.kde.KLauncher.xml)
-
-ki18n_wrap_ui(kcm_input_PART_SRCS kmousedlg.ui)
-qt5_add_dbus_interface(kcm_input_PART_SRCS ${klauncher_xml} klauncher_iface)
-
-add_library(kcm_input MODULE ${kcm_input_PART_SRCS})
-
-include_directories(${X11_X11_INCLUDE_PATH}
-                    ${X11_Xinput_INCLUDE_PATH}
-                    ${Evdev_INCLUDE_DIRS})
-
-target_link_libraries(kcm_input
-    Qt5::DBus
-    Qt5::X11Extras
-    KF5::KCMUtils
-    KF5::I18n
-    KF5::KIOCore
-    KF5::KIOWidgets
-    KF5::KDELibs4Support
-    ${X11_X11_LIB}
-    ${X11_Xinput_LIB}
-)
-if (X11_Xcursor_FOUND)
-   target_link_libraries(kcm_input ${X11_Xcursor_LIB})
-endif ()
-if (X11_Xfixes_FOUND)
-   target_link_libraries(kcm_input ${X11_Xfixes_LIB})
-endif ()
-
-install(TARGETS kcm_input  DESTINATION ${PLUGIN_INSTALL_DIR} )
-
-
 ########### install files ###############
 
-install( FILES mouse.desktop  DESTINATION  ${SERVICES_INSTALL_DIR} )
 install( FILES cursor_large_black.pcf.gz cursor_large_white.pcf.gz cursor_small_white.pcf.gz  DESTINATION  ${DATA_INSTALL_DIR}/kcminput )
 
--- plasma-desktop-5.4.2fix/kcms/touchpad/CMakeLists.txt	2015-10-18 10:35:53.630656827 +0800
+++ plasma-desktop-5.4.2/kcms/touchpad/CMakeLists.txt	2015-10-18 10:46:38.324693540 +0800
@@ -9,7 +9,6 @@
 set(CMAKE_INCLUDE_CURRENT_DIR true)
 set(TOUCHPAD_KCM_VERSION ${PROJECT_VERSION})
 
-add_subdirectory(src)
 add_subdirectory(icon)
 
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/l10n.cmake)
