diff --git a/applets/taskmanager/package/contents/ui/ToolTipDelegate.qml b/applets/taskmanager/package/contents/ui/ToolTipDelegate.qml
index f7084c7..e93610f 100644
--- a/applets/taskmanager/package/contents/ui/ToolTipDelegate.qml
+++ b/applets/taskmanager/package/contents/ui/ToolTipDelegate.qml
@@ -37,10 +37,6 @@ Column {
     property int preferredTextWidth: theme.mSize(theme.defaultFont).width * 30
     property int _s: units.largeSpacing / 2
 
-    Layout.minimumWidth: width
-    Layout.minimumHeight: height
-    Layout.maximumWidth: width
-    Layout.maximumHeight: height
     width: Math.max(windowRow.width, appLabelRow.width)  + _s
     height: childrenRect.height
     spacing: _s
