diff --git a/shell/CMakeLists.txt b/shell/CMakeLists.txt
index e17e4a3..f533802 100644
--- a/shell/CMakeLists.txt
+++ b/shell/CMakeLists.txt
@@ -67,6 +67,7 @@ target_link_libraries(plasmashell
  Qt5::Quick
  Qt5::DBus
  KF5::KIOCore
+ KF5::KIOWidgets
  KF5::WindowSystem
  KF5::Crash
  KF5::Plasma
diff --git a/shell/desktopview.cpp b/shell/desktopview.cpp
index 42a1f08..6923d63 100644
--- a/shell/desktopview.cpp
+++ b/shell/desktopview.cpp
@@ -26,6 +26,8 @@
 #include <QQmlContext>
 #include <QScreen>
 #include <qopenglshaderprogram.h>
+#include <QApplication>
+#include <QClipboard>
 
 #include <kwindowsystem.h>
 #include <klocalizedstring.h>
@@ -35,6 +37,8 @@
 #include <KWayland/Client/plasmashell.h>
 #include <KWayland/Client/surface.h>
 
+#include <KIOWidgets/KIO/PasteJob>
+
 DesktopView::DesktopView(Plasma::Corona *corona, QScreen *targetScreen)
     : PlasmaQuick::ContainmentView(corona, 0),
       m_windowType(Desktop),
@@ -215,6 +219,12 @@ void DesktopView::keyPressEvent(QKeyEvent *e)
 {
     ContainmentView::keyPressEvent(e);
 
+    if (e->matches(QKeySequence::Paste)) {
+        KIO::paste(QApplication::clipboard()->mimeData(), 
+            QUrl::fromLocalFile(QStandardPaths::writableLocation(QStandardPaths::DesktopLocation)));
+        return;
+    }
+
     // When a key is pressed on desktop when nothing else is active forward the key to krunner
     if (!e->modifiers() && !e->isAccepted()) {
         const QString text = e->text().trimmed();
