diff -Nur plasma-workspace.git/xembed-sni-proxy/sniproxy.cpp plasma-workspace/xembed-sni-proxy/sniproxy.cpp
--- plasma-workspace.git/xembed-sni-proxy/sniproxy.cpp	2015-11-26 16:43:42.289987877 +0800
+++ plasma-workspace/xembed-sni-proxy/sniproxy.cpp	2015-11-26 16:45:04.039986413 +0800
@@ -190,6 +190,8 @@
 SNIProxy::~SNIProxy()
 {
     QDBusConnection::disconnectFromBus(m_dbus.name());
+    auto c = QX11Info::connection();
+    xcb_destroy_window(c, m_containerWid); 
 }
 
 void SNIProxy::update()
@@ -401,9 +403,13 @@
         xcb_send_event(c, false, m_windowId, XCB_EVENT_MASK_BUTTON_RELEASE, (char *) event);
         free(event);
     }
+    //Do not move window down imediately, sometimes, tray requires double click.
+    QTimer::singleShot(1000, this, &SNIProxy::downContainer);
+}
 
-#ifndef VISUAL_DEBUG
+void SNIProxy::downContainer()
+{
     const uint32_t stackBelowData[] = {XCB_STACK_MODE_BELOW};
+    auto c = QX11Info::connection();
     xcb_configure_window(c, m_containerWid, XCB_CONFIG_WINDOW_STACK_MODE, stackBelowData);
-#endif
 }
diff -Nur plasma-workspace.git/xembed-sni-proxy/sniproxy.h plasma-workspace/xembed-sni-proxy/sniproxy.h
--- plasma-workspace.git/xembed-sni-proxy/sniproxy.h	2015-11-26 16:42:02.109989671 +0800
+++ plasma-workspace/xembed-sni-proxy/sniproxy.h	2015-11-26 16:43:56.382987625 +0800
@@ -46,6 +46,7 @@
     SNIProxy(xcb_window_t wid, QObject *parent=0);
     ~SNIProxy();
 
+    void downContainer();
     void update();
 
     /**
