diff --git a/lookandfeel/contents/splash/Splash.qml b/lookandfeel/contents/splash/Splash.qml
index fdeb1ea..9e2f2af 100644
--- a/lookandfeel/contents/splash/Splash.qml
+++ b/lookandfeel/contents/splash/Splash.qml
@@ -18,10 +18,11 @@
  */
 
 import QtQuick 2.2
+import QtAccountsService 1.0
 
 Image {
     id: root
-    source: "../components/artwork/background.png"
+    source: userAccount.background != "" ? userAccount.background : "../components/artwork/background.png"
 
     property int stage
 
@@ -30,6 +31,10 @@ Image {
             introAnimation.running = true
         }
     }
+
+    UserAccount {
+        id: userAccount
+    }
     Rectangle {
         id: topRect
         width: parent.width
@@ -37,7 +42,7 @@ Image {
         y: root.height
         color: "#4C000000"
         Image {
-            source: "images/kde.svgz"
+            source: userAccount.oslogo != "" ? userAccount.oslogo : "images/kde.svgz"
             anchors.centerIn: parent
             sourceSize.height: 128
             sourceSize.width: 128
