diff -Nur plasma-workspace-5.3.2/shell/shellcorona.cpp plasma-workspace-5.3.2n/shell/shellcorona.cpp
--- plasma-workspace-5.3.2/shell/shellcorona.cpp	2015-06-25 10:56:35.000000000 +0000
+++ plasma-workspace-5.3.2n/shell/shellcorona.cpp	2015-07-15 16:22:28.736282073 +0000
@@ -179,8 +179,10 @@
 ShellCorona::~ShellCorona()
 {
     qDeleteAll(m_views);
+    m_views.clear();
     qDeleteAll(containments());
     qDeleteAll(m_panelViews);
+    m_panelViews.clear();
 }
 
 KPackage::Package ShellCorona::lookAndFeelPackage()
