diff -Nur plasma-workspace-5.4.3/klipper/klipper.cpp plasma-workspace-5.4.3-n/klipper/klipper.cpp
--- plasma-workspace-5.4.3/klipper/klipper.cpp	2015-11-05 20:49:30.000000000 +0800
+++ plasma-workspace-5.4.3-n/klipper/klipper.cpp	2016-01-18 17:15:07.945188640 +0800
@@ -173,7 +173,7 @@
     m_editAction->setIcon(QIcon::fromTheme(QStringLiteral("document-properties")));
     m_editAction->setText(i18n("&Edit Contents..."));
     m_editAction->setVisible(m_mode == KlipperMode::Standalone);
-    KGlobalAccel::setGlobalShortcut(m_editAction, QKeySequence());
+//    KGlobalAccel::setGlobalShortcut(m_editAction, QKeySequence());
     connect(m_editAction, &QAction::triggered, this,
         [this]() {
             editData(m_history->first());
@@ -195,11 +195,11 @@
     // Cycle through history
     m_cycleNextAction = m_collection->addAction(QStringLiteral("cycleNextAction"));
     m_cycleNextAction->setText(i18n("Next History Item"));
-    KGlobalAccel::setGlobalShortcut(m_cycleNextAction, QKeySequence());
+//    KGlobalAccel::setGlobalShortcut(m_cycleNextAction, QKeySequence());
     connect(m_cycleNextAction, SIGNAL(triggered(bool)), SLOT(slotCycleNext()));
     m_cyclePrevAction = m_collection->addAction(QStringLiteral("cyclePrevAction"));
     m_cyclePrevAction->setText(i18n("Previous History Item"));
-    KGlobalAccel::setGlobalShortcut(m_cyclePrevAction, QKeySequence());
+//    KGlobalAccel::setGlobalShortcut(m_cyclePrevAction, QKeySequence());
     connect(m_cyclePrevAction, SIGNAL(triggered(bool)), SLOT(slotCyclePrev()));
 
     // Action to show Klipper popup on mouse position
