diff -Nur plasma-workspace-5.4.3/klipper/generalconfig.ui plasma-workspace-5.4.3n/klipper/generalconfig.ui
--- plasma-workspace-5.4.3/klipper/generalconfig.ui	2015-12-24 18:37:13.847721203 +0800
+++ plasma-workspace-5.4.3n/klipper/generalconfig.ui	2015-12-24 18:40:39.857740253 +0800
@@ -19,13 +19,6 @@
     </widget>
    </item>
    <item>
-    <widget class="QCheckBox" name="kcfg_PreventEmptyClipboard">
-     <property name="text">
-      <string>Prevent empty clipboard</string>
-     </property>
-    </widget>
-   </item>
-   <item>
     <widget class="QGroupBox" name="selection_group">
      <property name="title">
       <string>Selection and Clipboard</string>
