diff -Nur plasma-workspace-clean/plasma-workspace-5.4.3/klipper/generalconfig.ui plasma-workspace-5.4.3/klipper/generalconfig.ui
--- plasma-workspace-clean/plasma-workspace-5.4.3/klipper/generalconfig.ui	2015-11-05 20:49:30.000000000 +0800
+++ plasma-workspace-5.4.3/klipper/generalconfig.ui	2015-12-24 17:53:51.013480519 +0800
@@ -19,13 +19,6 @@
     </widget>
    </item>
    <item>
-    <widget class="QCheckBox" name="kcfg_PreventEmptyClipboard">
-     <property name="text">
-      <string>Prevent empty clipboard</string>
-     </property>
-    </widget>
-   </item>
-   <item>
     <widget class="QCheckBox" name="kcfg_IgnoreImages">
      <property name="text">
       <string>Ignore images</string>
