diff -Nur plasma-workspace-5.4.3/xembed-sni-proxy/sniproxy.cpp plasma-workspace-5.4.3n/xembed-sni-proxy/sniproxy.cpp
--- plasma-workspace-5.4.3/xembed-sni-proxy/sniproxy.cpp	2016-01-06 21:33:47.137945969 +0800
+++ plasma-workspace-5.4.3n/xembed-sni-proxy/sniproxy.cpp	2016-01-06 21:33:30.702946263 +0800
@@ -80,7 +80,9 @@
     QScopedPointer<xcb_get_geometry_reply_t, QScopedPointerPodDeleter>
         clientGeom(xcb_get_geometry_reply(c, cookie, Q_NULLPTR));
 
-    if (clientGeom && clientGeom->width > 0) {
+    if (clientGeom && clientGeom->width > 0 && 
+        !(clientGeom->width==683 && clientGeom->height==307) && 
+        !(clientGeom->width==1 && clientGeom->height==1)) {
     //create new SNI
     new StatusNotifierItemAdaptor(this);
     m_dbus.registerObject(QStringLiteral("/StatusNotifierItem"), this);
